// @devvit/protos/json/devvit/dev_portal/app_publish_request/app_publish_request.js
/**
 * #app_publish_request.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
export var AppPublishRequestStatus;
(function (AppPublishRequestStatus) {
    AppPublishRequestStatus[AppPublishRequestStatus["APPROVED"] = 0] = "APPROVED";
    AppPublishRequestStatus[AppPublishRequestStatus["DENIED"] = 1] = "DENIED";
    AppPublishRequestStatus[AppPublishRequestStatus["OBSOLETE"] = 2] = "OBSOLETE";
    AppPublishRequestStatus[AppPublishRequestStatus["PENDING"] = 3] = "PENDING";
    AppPublishRequestStatus[AppPublishRequestStatus["WITHDRAWN"] = 4] = "WITHDRAWN";
    AppPublishRequestStatus[AppPublishRequestStatus["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(AppPublishRequestStatus || (AppPublishRequestStatus = {}));
export var AppPublishRequestVisibility;
(function (AppPublishRequestVisibility) {
    AppPublishRequestVisibility[AppPublishRequestVisibility["PUBLIC"] = 0] = "PUBLIC";
    AppPublishRequestVisibility[AppPublishRequestVisibility["UNLISTED"] = 1] = "UNLISTED";
    AppPublishRequestVisibility[AppPublishRequestVisibility["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(AppPublishRequestVisibility || (AppPublishRequestVisibility = {}));
