// @devvit/protos/json/devvit/events/v1alpha/events.js
/**
 * #events.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
export var EventSource;
(function (EventSource) {
    EventSource[EventSource["UNKNOWN_EVENT_SOURCE"] = 0] = "UNKNOWN_EVENT_SOURCE";
    EventSource[EventSource["USER"] = 1] = "USER";
    EventSource[EventSource["ADMIN"] = 2] = "ADMIN";
    EventSource[EventSource["MODERATOR"] = 3] = "MODERATOR";
    EventSource[EventSource["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(EventSource || (EventSource = {}));
export var DeletionReason;
(function (DeletionReason) {
    DeletionReason[DeletionReason["UNSPECIFIED_DELETION_REASON"] = 0] = "UNSPECIFIED_DELETION_REASON";
    DeletionReason[DeletionReason["SPAM"] = 1] = "SPAM";
    DeletionReason[DeletionReason["LEGAL"] = 2] = "LEGAL";
    DeletionReason[DeletionReason["OTHER"] = 3] = "OTHER";
    DeletionReason[DeletionReason["UNKNOWN"] = 4] = "UNKNOWN";
    DeletionReason[DeletionReason["EXPLICIT_CONTENT"] = 5] = "EXPLICIT_CONTENT";
    DeletionReason[DeletionReason["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(DeletionReason || (DeletionReason = {}));
