// @devvit/protos/json/devvit/plugin/redditapi/common/common_msg.js
/**
 * #common_msg.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
/** Indicates who the request should be made on behalf of. Currently, only 'APP' or 'USER' types are supported. */
export var RunAs;
(function (RunAs) {
    RunAs[RunAs["APP"] = 0] = "APP";
    RunAs[RunAs["USER"] = 1] = "USER";
    RunAs[RunAs["UNSPECIFIED"] = 2] = "UNSPECIFIED";
    RunAs[RunAs["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(RunAs || (RunAs = {}));
export var GalleryMediaStatus;
(function (GalleryMediaStatus) {
    GalleryMediaStatus[GalleryMediaStatus["GALLERY_MEDIA_STATUS_UNSPECIFIED"] = 0] = "GALLERY_MEDIA_STATUS_UNSPECIFIED";
    GalleryMediaStatus[GalleryMediaStatus["GALLERY_MEDIA_STATUS_VALID"] = 1] = "GALLERY_MEDIA_STATUS_VALID";
    GalleryMediaStatus[GalleryMediaStatus["GALLERY_MEDIA_STATUS_FAILED"] = 2] = "GALLERY_MEDIA_STATUS_FAILED";
    GalleryMediaStatus[GalleryMediaStatus["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(GalleryMediaStatus || (GalleryMediaStatus = {}));
/** For methods with a request body, this specifies how the request body should be serialized. */
export var ApiClientConfig_BodyType;
(function (ApiClientConfig_BodyType) {
    ApiClientConfig_BodyType[ApiClientConfig_BodyType["NONE"] = 0] = "NONE";
    ApiClientConfig_BodyType[ApiClientConfig_BodyType["JSON_CAMEL"] = 1] = "JSON_CAMEL";
    ApiClientConfig_BodyType[ApiClientConfig_BodyType["JSON_SNAKE"] = 2] = "JSON_SNAKE";
    ApiClientConfig_BodyType[ApiClientConfig_BodyType["FORM_CAMEL"] = 3] = "FORM_CAMEL";
    ApiClientConfig_BodyType[ApiClientConfig_BodyType["FORM_SNAKE"] = 4] = "FORM_SNAKE";
    ApiClientConfig_BodyType[ApiClientConfig_BodyType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(ApiClientConfig_BodyType || (ApiClientConfig_BodyType = {}));
