// @devvit/protos/json/devvit/ui/block_kit/v1beta/enums.js
/**
 * #enums.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
export var BlockType;
(function (BlockType) {
    BlockType[BlockType["BLOCK_ROOT"] = 0] = "BLOCK_ROOT";
    BlockType[BlockType["BLOCK_STACK"] = 1] = "BLOCK_STACK";
    BlockType[BlockType["BLOCK_TEXT"] = 2] = "BLOCK_TEXT";
    BlockType[BlockType["BLOCK_BUTTON"] = 3] = "BLOCK_BUTTON";
    BlockType[BlockType["BLOCK_IMAGE"] = 4] = "BLOCK_IMAGE";
    BlockType[BlockType["BLOCK_SPACER"] = 5] = "BLOCK_SPACER";
    BlockType[BlockType["BLOCK_ICON"] = 6] = "BLOCK_ICON";
    BlockType[BlockType["BLOCK_AVATAR"] = 7] = "BLOCK_AVATAR";
    /** @deprecated */
    BlockType[BlockType["BLOCK_FULLSNOO"] = 8] = "BLOCK_FULLSNOO";
    /** @deprecated */
    BlockType[BlockType["BLOCK_ANIMATION"] = 9] = "BLOCK_ANIMATION";
    /** BLOCK_WEBVIEW - elements 100+ are privileged */
    BlockType[BlockType["BLOCK_WEBVIEW"] = 100] = "BLOCK_WEBVIEW";
    BlockType[BlockType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BlockType || (BlockType = {}));
export var BlockActionType;
(function (BlockActionType) {
    BlockActionType[BlockActionType["ACTION_CLICK"] = 0] = "ACTION_CLICK";
    BlockActionType[BlockActionType["ACTION_WEBVIEW"] = 100] = "ACTION_WEBVIEW";
    BlockActionType[BlockActionType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BlockActionType || (BlockActionType = {}));
export var BlockRenderEventType;
(function (BlockRenderEventType) {
    /** RENDER_INITIAL - First render of the UI on the user's device */
    BlockRenderEventType[BlockRenderEventType["RENDER_INITIAL"] = 0] = "RENDER_INITIAL";
    /** RENDER_USER_ACTION - Render triggered by a user action */
    BlockRenderEventType[BlockRenderEventType["RENDER_USER_ACTION"] = 1] = "RENDER_USER_ACTION";
    /** RENDER_EFFECT_EVENT - Render triggered as the result of an effect */
    BlockRenderEventType[BlockRenderEventType["RENDER_EFFECT_EVENT"] = 2] = "RENDER_EFFECT_EVENT";
    /** RENDER_CACHED - Render a version to be cached and used before the app can respond to RENDER_INITIAL */
    BlockRenderEventType[BlockRenderEventType["RENDER_CACHED"] = 3] = "RENDER_CACHED";
    BlockRenderEventType[BlockRenderEventType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BlockRenderEventType || (BlockRenderEventType = {}));
export var BlockStackDirection;
(function (BlockStackDirection) {
    BlockStackDirection[BlockStackDirection["STACK_HORIZONTAL"] = 0] = "STACK_HORIZONTAL";
    BlockStackDirection[BlockStackDirection["STACK_VERTICAL"] = 1] = "STACK_VERTICAL";
    BlockStackDirection[BlockStackDirection["STACK_DEPTH"] = 2] = "STACK_DEPTH";
    BlockStackDirection[BlockStackDirection["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BlockStackDirection || (BlockStackDirection = {}));
export var BlockBorderWidth;
(function (BlockBorderWidth) {
    BlockBorderWidth[BlockBorderWidth["BORDER_WIDTH_NONE"] = 0] = "BORDER_WIDTH_NONE";
    BlockBorderWidth[BlockBorderWidth["BORDER_WIDTH_THIN"] = 1] = "BORDER_WIDTH_THIN";
    BlockBorderWidth[BlockBorderWidth["BORDER_WIDTH_THICK"] = 2] = "BORDER_WIDTH_THICK";
    BlockBorderWidth[BlockBorderWidth["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BlockBorderWidth || (BlockBorderWidth = {}));
export var BlockVerticalAlignment;
(function (BlockVerticalAlignment) {
    BlockVerticalAlignment[BlockVerticalAlignment["ALIGN_TOP"] = 0] = "ALIGN_TOP";
    BlockVerticalAlignment[BlockVerticalAlignment["ALIGN_MIDDLE"] = 1] = "ALIGN_MIDDLE";
    BlockVerticalAlignment[BlockVerticalAlignment["ALIGN_BOTTOM"] = 2] = "ALIGN_BOTTOM";
    BlockVerticalAlignment[BlockVerticalAlignment["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BlockVerticalAlignment || (BlockVerticalAlignment = {}));
export var BlockHorizontalAlignment;
(function (BlockHorizontalAlignment) {
    BlockHorizontalAlignment[BlockHorizontalAlignment["ALIGN_START"] = 0] = "ALIGN_START";
    BlockHorizontalAlignment[BlockHorizontalAlignment["ALIGN_CENTER"] = 1] = "ALIGN_CENTER";
    BlockHorizontalAlignment[BlockHorizontalAlignment["ALIGN_END"] = 2] = "ALIGN_END";
    BlockHorizontalAlignment[BlockHorizontalAlignment["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BlockHorizontalAlignment || (BlockHorizontalAlignment = {}));
export var BlockPadding;
(function (BlockPadding) {
    BlockPadding[BlockPadding["PADDING_NONE"] = 0] = "PADDING_NONE";
    BlockPadding[BlockPadding["PADDING_XSMALL"] = 1] = "PADDING_XSMALL";
    BlockPadding[BlockPadding["PADDING_SMALL"] = 2] = "PADDING_SMALL";
    BlockPadding[BlockPadding["PADDING_MEDIUM"] = 3] = "PADDING_MEDIUM";
    BlockPadding[BlockPadding["PADDING_LARGE"] = 4] = "PADDING_LARGE";
    BlockPadding[BlockPadding["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BlockPadding || (BlockPadding = {}));
export var BlockGap;
(function (BlockGap) {
    BlockGap[BlockGap["GAP_NONE"] = 0] = "GAP_NONE";
    BlockGap[BlockGap["GAP_SMALL"] = 1] = "GAP_SMALL";
    BlockGap[BlockGap["GAP_MEDIUM"] = 2] = "GAP_MEDIUM";
    BlockGap[BlockGap["GAP_LARGE"] = 3] = "GAP_LARGE";
    BlockGap[BlockGap["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BlockGap || (BlockGap = {}));
export var BlockRadius;
(function (BlockRadius) {
    BlockRadius[BlockRadius["RADIUS_NONE"] = 0] = "RADIUS_NONE";
    BlockRadius[BlockRadius["RADIUS_SMALL"] = 1] = "RADIUS_SMALL";
    BlockRadius[BlockRadius["RADIUS_MEDIUM"] = 2] = "RADIUS_MEDIUM";
    BlockRadius[BlockRadius["RADIUS_LARGE"] = 3] = "RADIUS_LARGE";
    BlockRadius[BlockRadius["RADIUS_FULL"] = 100] = "RADIUS_FULL";
    BlockRadius[BlockRadius["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BlockRadius || (BlockRadius = {}));
export var BlockTextSize;
(function (BlockTextSize) {
    BlockTextSize[BlockTextSize["TEXT_SIZE_MEDIUM"] = 0] = "TEXT_SIZE_MEDIUM";
    BlockTextSize[BlockTextSize["TEXT_SIZE_XSMALL"] = 1] = "TEXT_SIZE_XSMALL";
    BlockTextSize[BlockTextSize["TEXT_SIZE_SMALL"] = 2] = "TEXT_SIZE_SMALL";
    BlockTextSize[BlockTextSize["TEXT_SIZE_LARGE"] = 3] = "TEXT_SIZE_LARGE";
    BlockTextSize[BlockTextSize["TEXT_SIZE_XLARGE"] = 4] = "TEXT_SIZE_XLARGE";
    BlockTextSize[BlockTextSize["TEXT_SIZE_XXLARGE"] = 5] = "TEXT_SIZE_XXLARGE";
    BlockTextSize[BlockTextSize["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BlockTextSize || (BlockTextSize = {}));
export var BlockTextWeight;
(function (BlockTextWeight) {
    BlockTextWeight[BlockTextWeight["TEXT_WEIGHT_REGULAR"] = 0] = "TEXT_WEIGHT_REGULAR";
    BlockTextWeight[BlockTextWeight["TEXT_WEIGHT_BOLD"] = 1] = "TEXT_WEIGHT_BOLD";
    BlockTextWeight[BlockTextWeight["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BlockTextWeight || (BlockTextWeight = {}));
export var BlockTextOutline;
(function (BlockTextOutline) {
    BlockTextOutline[BlockTextOutline["TEXT_OUTLINE_NONE"] = 0] = "TEXT_OUTLINE_NONE";
    BlockTextOutline[BlockTextOutline["TEXT_OUTLINE_THIN"] = 1] = "TEXT_OUTLINE_THIN";
    BlockTextOutline[BlockTextOutline["TEXT_OUTLINE_THICK"] = 2] = "TEXT_OUTLINE_THICK";
    BlockTextOutline[BlockTextOutline["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BlockTextOutline || (BlockTextOutline = {}));
export var BlockTextStyle;
(function (BlockTextStyle) {
    BlockTextStyle[BlockTextStyle["TEXT_STYLE_BODY"] = 0] = "TEXT_STYLE_BODY";
    BlockTextStyle[BlockTextStyle["TEXT_STYLE_METADATA"] = 1] = "TEXT_STYLE_METADATA";
    BlockTextStyle[BlockTextStyle["TEXT_STYLE_HEADING"] = 2] = "TEXT_STYLE_HEADING";
    BlockTextStyle[BlockTextStyle["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BlockTextStyle || (BlockTextStyle = {}));
export var BlockButtonSize;
(function (BlockButtonSize) {
    BlockButtonSize[BlockButtonSize["BUTTON_SIZE_MEDIUM"] = 0] = "BUTTON_SIZE_MEDIUM";
    BlockButtonSize[BlockButtonSize["BUTTON_SIZE_SMALL"] = 1] = "BUTTON_SIZE_SMALL";
    BlockButtonSize[BlockButtonSize["BUTTON_SIZE_LARGE"] = 2] = "BUTTON_SIZE_LARGE";
    BlockButtonSize[BlockButtonSize["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BlockButtonSize || (BlockButtonSize = {}));
export var BlockButtonAppearance;
(function (BlockButtonAppearance) {
    BlockButtonAppearance[BlockButtonAppearance["BUTTON_APPEARANCE_SECONDARY"] = 0] = "BUTTON_APPEARANCE_SECONDARY";
    BlockButtonAppearance[BlockButtonAppearance["BUTTON_APPEARANCE_PRIMARY"] = 1] = "BUTTON_APPEARANCE_PRIMARY";
    BlockButtonAppearance[BlockButtonAppearance["BUTTON_APPEARANCE_PLAIN"] = 2] = "BUTTON_APPEARANCE_PLAIN";
    BlockButtonAppearance[BlockButtonAppearance["BUTTON_APPEARANCE_BORDERED"] = 3] = "BUTTON_APPEARANCE_BORDERED";
    BlockButtonAppearance[BlockButtonAppearance["BUTTON_APPEARANCE_MEDIA"] = 4] = "BUTTON_APPEARANCE_MEDIA";
    BlockButtonAppearance[BlockButtonAppearance["BUTTON_APPEARANCE_DESTRUCTIVE"] = 5] = "BUTTON_APPEARANCE_DESTRUCTIVE";
    BlockButtonAppearance[BlockButtonAppearance["BUTTON_APPEARANCE_CAUTION"] = 6] = "BUTTON_APPEARANCE_CAUTION";
    BlockButtonAppearance[BlockButtonAppearance["BUTTON_APPEARANCE_SUCCESS"] = 7] = "BUTTON_APPEARANCE_SUCCESS";
    BlockButtonAppearance[BlockButtonAppearance["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BlockButtonAppearance || (BlockButtonAppearance = {}));
export var BlockImageResizeMode;
(function (BlockImageResizeMode) {
    BlockImageResizeMode[BlockImageResizeMode["IMAGE_RESIZE_FIT"] = 0] = "IMAGE_RESIZE_FIT";
    BlockImageResizeMode[BlockImageResizeMode["IMAGE_RESIZE_FILL"] = 1] = "IMAGE_RESIZE_FILL";
    BlockImageResizeMode[BlockImageResizeMode["IMAGE_RESIZE_COVER"] = 2] = "IMAGE_RESIZE_COVER";
    BlockImageResizeMode[BlockImageResizeMode["IMAGE_RESIZE_NONE"] = 3] = "IMAGE_RESIZE_NONE";
    BlockImageResizeMode[BlockImageResizeMode["IMAGE_RESIZE_SCALE_DOWN"] = 4] = "IMAGE_RESIZE_SCALE_DOWN";
    BlockImageResizeMode[BlockImageResizeMode["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BlockImageResizeMode || (BlockImageResizeMode = {}));
export var BlockSpacerSize;
(function (BlockSpacerSize) {
    BlockSpacerSize[BlockSpacerSize["SPACER_SMALL"] = 0] = "SPACER_SMALL";
    BlockSpacerSize[BlockSpacerSize["SPACER_XSMALL"] = 1] = "SPACER_XSMALL";
    BlockSpacerSize[BlockSpacerSize["SPACER_MEDIUM"] = 2] = "SPACER_MEDIUM";
    BlockSpacerSize[BlockSpacerSize["SPACER_LARGE"] = 3] = "SPACER_LARGE";
    BlockSpacerSize[BlockSpacerSize["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BlockSpacerSize || (BlockSpacerSize = {}));
export var BlockSpacerShape;
(function (BlockSpacerShape) {
    BlockSpacerShape[BlockSpacerShape["SPACER_INVISIBLE"] = 0] = "SPACER_INVISIBLE";
    BlockSpacerShape[BlockSpacerShape["SPACER_THIN"] = 1] = "SPACER_THIN";
    BlockSpacerShape[BlockSpacerShape["SPACER_SQUARE"] = 2] = "SPACER_SQUARE";
    BlockSpacerShape[BlockSpacerShape["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BlockSpacerShape || (BlockSpacerShape = {}));
export var BlockIconSize;
(function (BlockIconSize) {
    BlockIconSize[BlockIconSize["ICON_SIZE_MEDIUM"] = 0] = "ICON_SIZE_MEDIUM";
    BlockIconSize[BlockIconSize["ICON_SIZE_XSMALL"] = 1] = "ICON_SIZE_XSMALL";
    BlockIconSize[BlockIconSize["ICON_SIZE_SMALL"] = 2] = "ICON_SIZE_SMALL";
    BlockIconSize[BlockIconSize["ICON_SIZE_LARGE"] = 3] = "ICON_SIZE_LARGE";
    BlockIconSize[BlockIconSize["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BlockIconSize || (BlockIconSize = {}));
export var BlockAvatarFacing;
(function (BlockAvatarFacing) {
    BlockAvatarFacing[BlockAvatarFacing["AVATAR_FACING_LEFT"] = 0] = "AVATAR_FACING_LEFT";
    BlockAvatarFacing[BlockAvatarFacing["AVATAR_FACING_RIGHT"] = 1] = "AVATAR_FACING_RIGHT";
    BlockAvatarFacing[BlockAvatarFacing["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BlockAvatarFacing || (BlockAvatarFacing = {}));
export var BlockAvatarSize;
(function (BlockAvatarSize) {
    BlockAvatarSize[BlockAvatarSize["AVATAR_SIZE_SMALL"] = 0] = "AVATAR_SIZE_SMALL";
    BlockAvatarSize[BlockAvatarSize["AVATAR_SIZE_XXSMALL"] = 1] = "AVATAR_SIZE_XXSMALL";
    BlockAvatarSize[BlockAvatarSize["AVATAR_SIZE_XSMALL"] = 2] = "AVATAR_SIZE_XSMALL";
    BlockAvatarSize[BlockAvatarSize["AVATAR_SIZE_MEDIUM"] = 3] = "AVATAR_SIZE_MEDIUM";
    BlockAvatarSize[BlockAvatarSize["AVATAR_SIZE_LARGE"] = 4] = "AVATAR_SIZE_LARGE";
    BlockAvatarSize[BlockAvatarSize["AVATAR_SIZE_XLARGE"] = 5] = "AVATAR_SIZE_XLARGE";
    BlockAvatarSize[BlockAvatarSize["AVATAR_SIZE_XXLARGE"] = 6] = "AVATAR_SIZE_XXLARGE";
    BlockAvatarSize[BlockAvatarSize["AVATAR_SIZE_XXXLARGE"] = 7] = "AVATAR_SIZE_XXXLARGE";
    BlockAvatarSize[BlockAvatarSize["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BlockAvatarSize || (BlockAvatarSize = {}));
export var BlockAvatarBackground;
(function (BlockAvatarBackground) {
    BlockAvatarBackground[BlockAvatarBackground["AVATAR_BG_LIGHT"] = 0] = "AVATAR_BG_LIGHT";
    BlockAvatarBackground[BlockAvatarBackground["AVATAR_BG_DARK"] = 1] = "AVATAR_BG_DARK";
    BlockAvatarBackground[BlockAvatarBackground["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BlockAvatarBackground || (BlockAvatarBackground = {}));
/** @deprecated */
export var BlockFullSnooSize;
(function (BlockFullSnooSize) {
    BlockFullSnooSize[BlockFullSnooSize["FULLSNOO_SMALL"] = 0] = "FULLSNOO_SMALL";
    BlockFullSnooSize[BlockFullSnooSize["FULLSNOO_XSMALL"] = 1] = "FULLSNOO_XSMALL";
    BlockFullSnooSize[BlockFullSnooSize["FULLSNOO_MEDIUM"] = 2] = "FULLSNOO_MEDIUM";
    BlockFullSnooSize[BlockFullSnooSize["FULLSNOO_LARGE"] = 3] = "FULLSNOO_LARGE";
    BlockFullSnooSize[BlockFullSnooSize["FULLSNOO_XLARGE"] = 4] = "FULLSNOO_XLARGE";
    BlockFullSnooSize[BlockFullSnooSize["FULLSNOO_XXLARGE"] = 5] = "FULLSNOO_XXLARGE";
    BlockFullSnooSize[BlockFullSnooSize["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BlockFullSnooSize || (BlockFullSnooSize = {}));
/** @deprecated */
export var BlockAnimationType;
(function (BlockAnimationType) {
    BlockAnimationType[BlockAnimationType["ANIM_LOTTIE"] = 0] = "ANIM_LOTTIE";
    BlockAnimationType[BlockAnimationType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BlockAnimationType || (BlockAnimationType = {}));
/** @deprecated */
export var BlockAnimationLoopMode;
(function (BlockAnimationLoopMode) {
    BlockAnimationLoopMode[BlockAnimationLoopMode["ANIM_LOOP_REPEAT"] = 0] = "ANIM_LOOP_REPEAT";
    BlockAnimationLoopMode[BlockAnimationLoopMode["ANIM_LOOP_BOUNCE"] = 1] = "ANIM_LOOP_BOUNCE";
    BlockAnimationLoopMode[BlockAnimationLoopMode["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BlockAnimationLoopMode || (BlockAnimationLoopMode = {}));
/** @deprecated */
export var BlockAnimationDirection;
(function (BlockAnimationDirection) {
    BlockAnimationDirection[BlockAnimationDirection["ANIM_DIR_FORWARD"] = 0] = "ANIM_DIR_FORWARD";
    BlockAnimationDirection[BlockAnimationDirection["ANIM_DIR_BACKWARD"] = 1] = "ANIM_DIR_BACKWARD";
    BlockAnimationDirection[BlockAnimationDirection["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BlockAnimationDirection || (BlockAnimationDirection = {}));
export var BlockSizeUnit;
(function (BlockSizeUnit) {
    BlockSizeUnit[BlockSizeUnit["SIZE_UNIT_PERCENT"] = 0] = "SIZE_UNIT_PERCENT";
    BlockSizeUnit[BlockSizeUnit["SIZE_UNIT_PIXELS"] = 1] = "SIZE_UNIT_PIXELS";
    BlockSizeUnit[BlockSizeUnit["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BlockSizeUnit || (BlockSizeUnit = {}));
export var BlockTextOverflow;
(function (BlockTextOverflow) {
    BlockTextOverflow[BlockTextOverflow["TEXT_OVERFLOW_CLIP"] = 0] = "TEXT_OVERFLOW_CLIP";
    BlockTextOverflow[BlockTextOverflow["TEXT_OVERFLOW_ELLIPSE"] = 1] = "TEXT_OVERFLOW_ELLIPSE";
    BlockTextOverflow[BlockTextOverflow["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BlockTextOverflow || (BlockTextOverflow = {}));
