// @devvit/protos/json/reddit/devvit/common/v1/app.js
/**
 * #app.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
export var AppVisibility;
(function (AppVisibility) {
    AppVisibility[AppVisibility["PUBLIC"] = 0] = "PUBLIC";
    AppVisibility[AppVisibility["PRIVATE"] = 1] = "PRIVATE";
    AppVisibility[AppVisibility["UNLISTED"] = 2] = "UNLISTED";
    AppVisibility[AppVisibility["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(AppVisibility || (AppVisibility = {}));
/** App render expectation. */
export var RenderVersion;
(function (RenderVersion) {
    RenderVersion[RenderVersion["RENDER_VERSION_UNSPECIFIED"] = 0] = "RENDER_VERSION_UNSPECIFIED";
    /** NO_DEVVIT_JSON - `devvit.yaml` Blocks app. */
    RenderVersion[RenderVersion["NO_DEVVIT_JSON"] = 256] = "NO_DEVVIT_JSON";
    /** WEBBIT_BLOCKS - Classic or hybrid Webbit app. */
    RenderVersion[RenderVersion["WEBBIT_BLOCKS"] = 512] = "WEBBIT_BLOCKS";
    RenderVersion[RenderVersion["WEBBIT_PURE_NO_ENTRYPOINT"] = 1024] = "WEBBIT_PURE_NO_ENTRYPOINT";
    /** WEBBIT_PURE_NO_INLINE - Entry exists but isn't explicitly inline. */
    RenderVersion[RenderVersion["WEBBIT_PURE_NO_INLINE"] = 1040] = "WEBBIT_PURE_NO_INLINE";
    /** WEBBIT_PURE_INLINE - Entry exists and is explicitly inline. */
    RenderVersion[RenderVersion["WEBBIT_PURE_INLINE"] = 1056] = "WEBBIT_PURE_INLINE";
    RenderVersion[RenderVersion["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(RenderVersion || (RenderVersion = {}));
