// @devvit/protos/json/reddit/devvit/app_permission/v1/app_permission.js
/**
 * #app_permission.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
/** This enum is used to represent the consent status of an app permission. */
export var ConsentStatus;
(function (ConsentStatus) {
    /** CONSENT_STATUS_UNKNOWN - The consent status is unknown, which means it has not been set or is not applicable. */
    ConsentStatus[ConsentStatus["CONSENT_STATUS_UNKNOWN"] = 0] = "CONSENT_STATUS_UNKNOWN";
    /** REVOKED - The user has explicitly denied consent for the app permissions. */
    ConsentStatus[ConsentStatus["REVOKED"] = 1] = "REVOKED";
    /** GRANTED - The user has granted consent for the app permissions. */
    ConsentStatus[ConsentStatus["GRANTED"] = 2] = "GRANTED";
    ConsentStatus[ConsentStatus["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(ConsentStatus || (ConsentStatus = {}));
/** This enum is used to represent the scopes of permissions that can be granted to an app. */
export var Scope;
(function (Scope) {
    /** SCOPE_UNKNOWN - The scope is unknown, which means it has not been set or is not applicable. */
    Scope[Scope["SCOPE_UNKNOWN"] = 0] = "SCOPE_UNKNOWN";
    /** SUBMIT_POST - Allows the app to submit posts on behalf of the user. */
    Scope[Scope["SUBMIT_POST"] = 1] = "SUBMIT_POST";
    /** SUBMIT_COMMENT - Allows the app to submit comments on behalf of the user. */
    Scope[Scope["SUBMIT_COMMENT"] = 2] = "SUBMIT_COMMENT";
    /** SUBSCRIBE_TO_SUBREDDIT - Allows the app to subscribe the user to a subreddit. */
    Scope[Scope["SUBSCRIBE_TO_SUBREDDIT"] = 3] = "SUBSCRIBE_TO_SUBREDDIT";
    Scope[Scope["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(Scope || (Scope = {}));
