// @devvit/protos/types/devvit/plugin_svc/notifications/template_svc.js
/**
 * #template_svc.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
import _m0 from "protobufjs/minimal.js";
import { Empty } from "../../../google/protobuf/empty.js";
import { GetTemplatesResponse, RegisterTemplatesRequest, RegisterTemplatesResponse } from "./template_msg.js";
export const TemplateServiceName = "devvit.plugin_svc.notifications.Template";
export class TemplateClientImpl {
    constructor(rpc, opts) {
        this.service = opts?.service || TemplateServiceName;
        this.rpc = rpc;
        this.RegisterTemplates = this.RegisterTemplates.bind(this);
        this.GetTemplates = this.GetTemplates.bind(this);
    }
    RegisterTemplates(request, metadata) {
        const data = RegisterTemplatesRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "RegisterTemplates", data, metadata);
        return promise.then((data) => RegisterTemplatesResponse.decode(_m0.Reader.create(data)));
    }
    GetTemplates(request, metadata) {
        const data = Empty.encode(request).finish();
        const promise = this.rpc.request(this.service, "GetTemplates", data, metadata);
        return promise.then((data) => GetTemplatesResponse.decode(_m0.Reader.create(data)));
    }
}
export const TemplateDefinition = {
    name: "Template",
    fullName: "devvit.plugin_svc.notifications.Template",
    methods: {
        /** Register push notification templates for the app */
        registerTemplates: {
            name: "RegisterTemplates",
            requestType: RegisterTemplatesRequest,
            requestStream: false,
            responseType: RegisterTemplatesResponse,
            responseStream: false,
            options: {},
        },
        /** Get registered push notification templates for the app */
        getTemplates: {
            name: "GetTemplates",
            requestType: Empty,
            requestStream: false,
            responseType: GetTemplatesResponse,
            responseStream: false,
            options: {},
        },
    },
};
