// @devvit/protos/types/devvit/plugin_svc/notifications/template_svc.twirp.js
import { TwirpServer, TwirpError, TwirpErrorCode, TwirpContentType, chainInterceptors, } from 'twirp-ts';
import { RegisterTemplatesRequest, RegisterTemplatesResponse, GetTemplatesResponse, } from './template_msg.js';
import { Empty } from '../../../google/protobuf/empty.js';
export var TemplateMethod;
(function (TemplateMethod) {
    TemplateMethod["RegisterTemplates"] = "RegisterTemplates";
    TemplateMethod["GetTemplates"] = "GetTemplates";
})(TemplateMethod || (TemplateMethod = {}));
export const TemplateMethodList = [
    TemplateMethod.RegisterTemplates,
    TemplateMethod.GetTemplates,
];
export function createTemplateServer(service) {
    return new TwirpServer({
        service,
        packageName: 'devvit.plugin_svc.notifications',
        serviceName: 'Template',
        methodList: TemplateMethodList,
        matchRoute: matchTemplateRoute,
    });
}
function matchTemplateRoute(method, events) {
    switch (method) {
        case 'RegisterTemplates':
            return async (ctx, service, data, interceptors) => {
                ctx = { ...ctx, methodName: 'RegisterTemplates' };
                await events.onMatch(ctx);
                return handleTemplateRegisterTemplatesRequest(ctx, service, data, interceptors);
            };
        case 'GetTemplates':
            return async (ctx, service, data, interceptors) => {
                ctx = { ...ctx, methodName: 'GetTemplates' };
                await events.onMatch(ctx);
                return handleTemplateGetTemplatesRequest(ctx, service, data, interceptors);
            };
        default:
            events.onNotFound();
            const msg = `no handler found`;
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
function handleTemplateRegisterTemplatesRequest(ctx, service, data, interceptors) {
    switch (ctx.contentType) {
        case TwirpContentType.JSON:
            return handleTemplateRegisterTemplatesJSON(ctx, service, data, interceptors);
        case TwirpContentType.Protobuf:
            return handleTemplateRegisterTemplatesProtobuf(ctx, service, data, interceptors);
        default:
            const msg = 'unexpected Content-Type';
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
function handleTemplateGetTemplatesRequest(ctx, service, data, interceptors) {
    switch (ctx.contentType) {
        case TwirpContentType.JSON:
            return handleTemplateGetTemplatesJSON(ctx, service, data, interceptors);
        case TwirpContentType.Protobuf:
            return handleTemplateGetTemplatesProtobuf(ctx, service, data, interceptors);
        default:
            const msg = 'unexpected Content-Type';
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
async function handleTemplateRegisterTemplatesJSON(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        const body = JSON.parse(data.toString() || '{}');
        request = RegisterTemplatesRequest.fromJSON(body);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the json request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.RegisterTemplates(ctx, inputReq);
        });
    }
    else {
        response = await service.RegisterTemplates(ctx, request);
    }
    return JSON.stringify(RegisterTemplatesResponse.toJSON(response));
}
async function handleTemplateGetTemplatesJSON(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        const body = JSON.parse(data.toString() || '{}');
        request = Empty.fromJSON(body);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the json request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.GetTemplates(ctx, inputReq);
        });
    }
    else {
        response = await service.GetTemplates(ctx, request);
    }
    return JSON.stringify(GetTemplatesResponse.toJSON(response));
}
async function handleTemplateRegisterTemplatesProtobuf(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        request = RegisterTemplatesRequest.decode(data);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the protobuf request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.RegisterTemplates(ctx, inputReq);
        });
    }
    else {
        response = await service.RegisterTemplates(ctx, request);
    }
    return Buffer.from(RegisterTemplatesResponse.encode(response).finish());
}
async function handleTemplateGetTemplatesProtobuf(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        request = Empty.decode(data);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the protobuf request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.GetTemplates(ctx, inputReq);
        });
    }
    else {
        response = await service.GetTemplates(ctx, request);
    }
    return Buffer.from(GetTemplatesResponse.encode(response).finish());
}
