// @devvit/protos/types/devvit/plugin_svc/notifications/template_msg.js
/**
 * #template_msg.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
import _m0 from "protobufjs/minimal.js";
import { messageTypeRegistry } from "../../../typeRegistry.js";
function createBasePushNotificationTemplate() {
    return { templateId: "", titleTemplate: "", bodyTemplate: "" };
}
export const PushNotificationTemplate = {
    $type: "devvit.plugin_svc.notifications.PushNotificationTemplate",
    encode(message, writer = _m0.Writer.create()) {
        if (message.templateId !== "") {
            writer.uint32(10).string(message.templateId);
        }
        if (message.titleTemplate !== "") {
            writer.uint32(18).string(message.titleTemplate);
        }
        if (message.bodyTemplate !== "") {
            writer.uint32(26).string(message.bodyTemplate);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePushNotificationTemplate();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.templateId = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.titleTemplate = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.bodyTemplate = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            templateId: isSet(object.templateId) ? globalThis.String(object.templateId) : "",
            titleTemplate: isSet(object.titleTemplate) ? globalThis.String(object.titleTemplate) : "",
            bodyTemplate: isSet(object.bodyTemplate) ? globalThis.String(object.bodyTemplate) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.templateId !== "") {
            obj.templateId = message.templateId;
        }
        if (message.titleTemplate !== "") {
            obj.titleTemplate = message.titleTemplate;
        }
        if (message.bodyTemplate !== "") {
            obj.bodyTemplate = message.bodyTemplate;
        }
        return obj;
    },
    create(base) {
        return PushNotificationTemplate.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePushNotificationTemplate();
        message.templateId = object.templateId ?? "";
        message.titleTemplate = object.titleTemplate ?? "";
        message.bodyTemplate = object.bodyTemplate ?? "";
        return message;
    },
};
messageTypeRegistry.set(PushNotificationTemplate.$type, PushNotificationTemplate);
function createBaseRegisterTemplatesRequest() {
    return { templates: [] };
}
export const RegisterTemplatesRequest = {
    $type: "devvit.plugin_svc.notifications.RegisterTemplatesRequest",
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.templates) {
            PushNotificationTemplate.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRegisterTemplatesRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.templates.push(PushNotificationTemplate.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            templates: globalThis.Array.isArray(object?.templates)
                ? object.templates.map((e) => PushNotificationTemplate.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.templates?.length) {
            obj.templates = message.templates.map((e) => PushNotificationTemplate.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return RegisterTemplatesRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseRegisterTemplatesRequest();
        message.templates = object.templates?.map((e) => PushNotificationTemplate.fromPartial(e)) || [];
        return message;
    },
};
messageTypeRegistry.set(RegisterTemplatesRequest.$type, RegisterTemplatesRequest);
function createBasePushNotificationTemplateErrors() {
    return { templateId: "", errorMessage: "" };
}
export const PushNotificationTemplateErrors = {
    $type: "devvit.plugin_svc.notifications.PushNotificationTemplateErrors",
    encode(message, writer = _m0.Writer.create()) {
        if (message.templateId !== "") {
            writer.uint32(10).string(message.templateId);
        }
        if (message.errorMessage !== "") {
            writer.uint32(18).string(message.errorMessage);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePushNotificationTemplateErrors();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.templateId = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.errorMessage = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            templateId: isSet(object.templateId) ? globalThis.String(object.templateId) : "",
            errorMessage: isSet(object.errorMessage) ? globalThis.String(object.errorMessage) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.templateId !== "") {
            obj.templateId = message.templateId;
        }
        if (message.errorMessage !== "") {
            obj.errorMessage = message.errorMessage;
        }
        return obj;
    },
    create(base) {
        return PushNotificationTemplateErrors.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePushNotificationTemplateErrors();
        message.templateId = object.templateId ?? "";
        message.errorMessage = object.errorMessage ?? "";
        return message;
    },
};
messageTypeRegistry.set(PushNotificationTemplateErrors.$type, PushNotificationTemplateErrors);
function createBaseRegisterTemplatesResponse() {
    return { success: false, errors: [] };
}
export const RegisterTemplatesResponse = {
    $type: "devvit.plugin_svc.notifications.RegisterTemplatesResponse",
    encode(message, writer = _m0.Writer.create()) {
        if (message.success !== false) {
            writer.uint32(8).bool(message.success);
        }
        for (const v of message.errors) {
            PushNotificationTemplateErrors.encode(v, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRegisterTemplatesResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.success = reader.bool();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.errors.push(PushNotificationTemplateErrors.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            success: isSet(object.success) ? globalThis.Boolean(object.success) : false,
            errors: globalThis.Array.isArray(object?.errors)
                ? object.errors.map((e) => PushNotificationTemplateErrors.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.success !== false) {
            obj.success = message.success;
        }
        if (message.errors?.length) {
            obj.errors = message.errors.map((e) => PushNotificationTemplateErrors.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return RegisterTemplatesResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseRegisterTemplatesResponse();
        message.success = object.success ?? false;
        message.errors = object.errors?.map((e) => PushNotificationTemplateErrors.fromPartial(e)) || [];
        return message;
    },
};
messageTypeRegistry.set(RegisterTemplatesResponse.$type, RegisterTemplatesResponse);
function createBaseGetTemplatesResponse() {
    return { templates: [] };
}
export const GetTemplatesResponse = {
    $type: "devvit.plugin_svc.notifications.GetTemplatesResponse",
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.templates) {
            PushNotificationTemplate.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetTemplatesResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.templates.push(PushNotificationTemplate.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            templates: globalThis.Array.isArray(object?.templates)
                ? object.templates.map((e) => PushNotificationTemplate.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.templates?.length) {
            obj.templates = message.templates.map((e) => PushNotificationTemplate.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return GetTemplatesResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetTemplatesResponse();
        message.templates = object.templates?.map((e) => PushNotificationTemplate.fromPartial(e)) || [];
        return message;
    },
};
messageTypeRegistry.set(GetTemplatesResponse.$type, GetTemplatesResponse);
function isSet(value) {
    return value !== null && value !== undefined;
}
