// @devvit/protos/json/devvit/reddit/v2alpha/modaction.js
/**
 * #modaction.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
export var ModActionType;
(function (ModActionType) {
    ModActionType[ModActionType["UNKNOWN"] = 0] = "UNKNOWN";
    ModActionType[ModActionType["REMOVE_POST"] = 1] = "REMOVE_POST";
    ModActionType[ModActionType["REMOVE_COMMENT"] = 2] = "REMOVE_COMMENT";
    ModActionType[ModActionType["APPROVE_POST"] = 3] = "APPROVE_POST";
    ModActionType[ModActionType["APPROVE_COMMENT"] = 4] = "APPROVE_COMMENT";
    ModActionType[ModActionType["EDIT_POST_FLAIR"] = 5] = "EDIT_POST_FLAIR";
    ModActionType[ModActionType["EDIT_USER_FLAIR"] = 6] = "EDIT_USER_FLAIR";
    ModActionType[ModActionType["LOCK_POST"] = 7] = "LOCK_POST";
    ModActionType[ModActionType["LOCK_COMMENT"] = 8] = "LOCK_COMMENT";
    ModActionType[ModActionType["SPAM_POST"] = 9] = "SPAM_POST";
    ModActionType[ModActionType["SPAM_COMMENT"] = 10] = "SPAM_COMMENT";
    ModActionType[ModActionType["DISTINGUISH_POST"] = 11] = "DISTINGUISH_POST";
    ModActionType[ModActionType["DISTINGUISH_COMMENT"] = 12] = "DISTINGUISH_COMMENT";
    ModActionType[ModActionType["STICKY_POST"] = 13] = "STICKY_POST";
    ModActionType[ModActionType["STICKY_COMMENT"] = 14] = "STICKY_COMMENT";
    ModActionType[ModActionType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(ModActionType || (ModActionType = {}));
