// @devvit/protos/json/devvit/reddit/v2alpha/subredditv2.js
/**
 * #subredditv2.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
export var SubredditType;
(function (SubredditType) {
    SubredditType[SubredditType["ARCHIVED"] = 0] = "ARCHIVED";
    SubredditType[SubredditType["EMPLOYEES_ONLY"] = 1] = "EMPLOYEES_ONLY";
    SubredditType[SubredditType["GOLD_ONLY"] = 2] = "GOLD_ONLY";
    SubredditType[SubredditType["GOLD_RESTRICTED"] = 3] = "GOLD_RESTRICTED";
    SubredditType[SubredditType["PRIVATE"] = 4] = "PRIVATE";
    SubredditType[SubredditType["PUBLIC"] = 5] = "PUBLIC";
    SubredditType[SubredditType["RESTRICTED"] = 6] = "RESTRICTED";
    SubredditType[SubredditType["USER"] = 7] = "USER";
    SubredditType[SubredditType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(SubredditType || (SubredditType = {}));
/** used with data-api */
export var SubredditRating;
(function (SubredditRating) {
    SubredditRating[SubredditRating["UNKNOWN_SUBREDDIT_RATING"] = 0] = "UNKNOWN_SUBREDDIT_RATING";
    /** E - everyone */
    SubredditRating[SubredditRating["E"] = 1] = "E";
    /** M1 - mature 1 */
    SubredditRating[SubredditRating["M1"] = 2] = "M1";
    /** M2 - mature 2 */
    SubredditRating[SubredditRating["M2"] = 3] = "M2";
    /** D - high-risk drug use */
    SubredditRating[SubredditRating["D"] = 4] = "D";
    /** V - violence & gore */
    SubredditRating[SubredditRating["V"] = 5] = "V";
    /** X - sexually explicit */
    SubredditRating[SubredditRating["X"] = 6] = "X";
    SubredditRating[SubredditRating["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(SubredditRating || (SubredditRating = {}));
