// @devvit/protos/json/devvit/dev_portal/payments/payments_verification.js
/**
 * #payments_verification.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
/**
 * The verification status of a devvit developer. This shows whether or not a user can receive payments
 * and their state in the verification flow. See design doc here:
 * https://docs.google.com/document/d/1HuAfNB_26jnNu7gGDHk-vafzgpnoNaFqHoumE2C40cA/edit?tab=t.0#bookmark=id.hm6xb18pqjvh
 */
export var PaymentsVerificationStatus;
(function (PaymentsVerificationStatus) {
    PaymentsVerificationStatus[PaymentsVerificationStatus["UNKNOWN"] = 0] = "UNKNOWN";
    /** VERIFICATION_NOT_STARTED - User has not begun the verification process and is permitted to. */
    PaymentsVerificationStatus[PaymentsVerificationStatus["VERIFICATION_NOT_STARTED"] = 1] = "VERIFICATION_NOT_STARTED";
    /** VERIFICATION_PENDING - User has begun the verification process. */
    PaymentsVerificationStatus[PaymentsVerificationStatus["VERIFICATION_PENDING"] = 2] = "VERIFICATION_PENDING";
    /**
     * VERIFICATION_CAN_NOT_START - User is not eligible to receive payments. This may be because they're ineligible to apply for payments
     * or because they applied and were rejected.
     */
    PaymentsVerificationStatus[PaymentsVerificationStatus["VERIFICATION_CAN_NOT_START"] = 3] = "VERIFICATION_CAN_NOT_START";
    /** VERIFICATION_SUCCESS - Verification has been completed and the user is able to receive payments. */
    PaymentsVerificationStatus[PaymentsVerificationStatus["VERIFICATION_SUCCESS"] = 4] = "VERIFICATION_SUCCESS";
    PaymentsVerificationStatus[PaymentsVerificationStatus["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(PaymentsVerificationStatus || (PaymentsVerificationStatus = {}));
