// @devvit/protos/types/devvit/plugin/notifications/notifications_svc.twirp-client.js
import { Empty } from '../../../google/protobuf/empty.js';
import { OptInCurrentUserResponse, OptOutCurrentUserResponse, ListOptedInUsersRequest, ListOptedInUsersResponse, IsOptedInRequest, IsOptedInResponse, } from './optin_msg.js';
import { EnqueueRequest, EnqueueResponse } from './pushnotif_msg.js';
import { ShowGamesDrawerBadgeRequest, ShowGamesDrawerBadgeResponse, DismissGamesDrawerBadgeRequest, DismissGamesDrawerBadgeResponse, GetGamesDrawerBadgeStatusRequest, GetGamesDrawerBadgeStatusResponse, } from './gamesdrawerbadge_msg.js';
export class NotificationsClientJSON {
    constructor(rpc) {
        this.rpc = rpc;
        this.OptInCurrentUser.bind(this);
        this.OptOutCurrentUser.bind(this);
        this.ListOptedInUsers.bind(this);
        this.IsOptedIn.bind(this);
        this.Enqueue.bind(this);
        this.ShowGamesDrawerBadge.bind(this);
        this.DismissGamesDrawerBadge.bind(this);
        this.GetGamesDrawerBadgeStatus.bind(this);
    }
    OptInCurrentUser(request) {
        const data = Empty.toJSON(request);
        const promise = this.rpc.request('devvit.plugin.notifications.Notifications', 'OptInCurrentUser', 'application/json', data);
        return promise.then((data) => OptInCurrentUserResponse.fromJSON(data));
    }
    OptOutCurrentUser(request) {
        const data = Empty.toJSON(request);
        const promise = this.rpc.request('devvit.plugin.notifications.Notifications', 'OptOutCurrentUser', 'application/json', data);
        return promise.then((data) => OptOutCurrentUserResponse.fromJSON(data));
    }
    ListOptedInUsers(request) {
        const data = ListOptedInUsersRequest.toJSON(request);
        const promise = this.rpc.request('devvit.plugin.notifications.Notifications', 'ListOptedInUsers', 'application/json', data);
        return promise.then((data) => ListOptedInUsersResponse.fromJSON(data));
    }
    IsOptedIn(request) {
        const data = IsOptedInRequest.toJSON(request);
        const promise = this.rpc.request('devvit.plugin.notifications.Notifications', 'IsOptedIn', 'application/json', data);
        return promise.then((data) => IsOptedInResponse.fromJSON(data));
    }
    Enqueue(request) {
        const data = EnqueueRequest.toJSON(request);
        const promise = this.rpc.request('devvit.plugin.notifications.Notifications', 'Enqueue', 'application/json', data);
        return promise.then((data) => EnqueueResponse.fromJSON(data));
    }
    ShowGamesDrawerBadge(request) {
        const data = ShowGamesDrawerBadgeRequest.toJSON(request);
        const promise = this.rpc.request('devvit.plugin.notifications.Notifications', 'ShowGamesDrawerBadge', 'application/json', data);
        return promise.then((data) => ShowGamesDrawerBadgeResponse.fromJSON(data));
    }
    DismissGamesDrawerBadge(request) {
        const data = DismissGamesDrawerBadgeRequest.toJSON(request);
        const promise = this.rpc.request('devvit.plugin.notifications.Notifications', 'DismissGamesDrawerBadge', 'application/json', data);
        return promise.then((data) => DismissGamesDrawerBadgeResponse.fromJSON(data));
    }
    GetGamesDrawerBadgeStatus(request) {
        const data = GetGamesDrawerBadgeStatusRequest.toJSON(request);
        const promise = this.rpc.request('devvit.plugin.notifications.Notifications', 'GetGamesDrawerBadgeStatus', 'application/json', data);
        return promise.then((data) => GetGamesDrawerBadgeStatusResponse.fromJSON(data));
    }
}
export class NotificationsClientProtobuf {
    constructor(rpc) {
        this.rpc = rpc;
        this.OptInCurrentUser.bind(this);
        this.OptOutCurrentUser.bind(this);
        this.ListOptedInUsers.bind(this);
        this.IsOptedIn.bind(this);
        this.Enqueue.bind(this);
        this.ShowGamesDrawerBadge.bind(this);
        this.DismissGamesDrawerBadge.bind(this);
        this.GetGamesDrawerBadgeStatus.bind(this);
    }
    OptInCurrentUser(request) {
        const data = Empty.encode(request).finish();
        const promise = this.rpc.request('devvit.plugin.notifications.Notifications', 'OptInCurrentUser', 'application/protobuf', data);
        return promise.then((data) => OptInCurrentUserResponse.decode(data));
    }
    OptOutCurrentUser(request) {
        const data = Empty.encode(request).finish();
        const promise = this.rpc.request('devvit.plugin.notifications.Notifications', 'OptOutCurrentUser', 'application/protobuf', data);
        return promise.then((data) => OptOutCurrentUserResponse.decode(data));
    }
    ListOptedInUsers(request) {
        const data = ListOptedInUsersRequest.encode(request).finish();
        const promise = this.rpc.request('devvit.plugin.notifications.Notifications', 'ListOptedInUsers', 'application/protobuf', data);
        return promise.then((data) => ListOptedInUsersResponse.decode(data));
    }
    IsOptedIn(request) {
        const data = IsOptedInRequest.encode(request).finish();
        const promise = this.rpc.request('devvit.plugin.notifications.Notifications', 'IsOptedIn', 'application/protobuf', data);
        return promise.then((data) => IsOptedInResponse.decode(data));
    }
    Enqueue(request) {
        const data = EnqueueRequest.encode(request).finish();
        const promise = this.rpc.request('devvit.plugin.notifications.Notifications', 'Enqueue', 'application/protobuf', data);
        return promise.then((data) => EnqueueResponse.decode(data));
    }
    ShowGamesDrawerBadge(request) {
        const data = ShowGamesDrawerBadgeRequest.encode(request).finish();
        const promise = this.rpc.request('devvit.plugin.notifications.Notifications', 'ShowGamesDrawerBadge', 'application/protobuf', data);
        return promise.then((data) => ShowGamesDrawerBadgeResponse.decode(data));
    }
    DismissGamesDrawerBadge(request) {
        const data = DismissGamesDrawerBadgeRequest.encode(request).finish();
        const promise = this.rpc.request('devvit.plugin.notifications.Notifications', 'DismissGamesDrawerBadge', 'application/protobuf', data);
        return promise.then((data) => DismissGamesDrawerBadgeResponse.decode(data));
    }
    GetGamesDrawerBadgeStatus(request) {
        const data = GetGamesDrawerBadgeStatusRequest.encode(request).finish();
        const promise = this.rpc.request('devvit.plugin.notifications.Notifications', 'GetGamesDrawerBadgeStatus', 'application/protobuf', data);
        return promise.then((data) => GetGamesDrawerBadgeStatusResponse.decode(data));
    }
}
