// @devvit/protos/types/devvit/plugin/notifications/notifications_svc.js
/**
 * #notifications_svc.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
import _m0 from "protobufjs/minimal.js";
import { Empty } from "../../../google/protobuf/empty.js";
import { DismissGamesDrawerBadgeRequest, DismissGamesDrawerBadgeResponse, GetGamesDrawerBadgeStatusRequest, GetGamesDrawerBadgeStatusResponse, ShowGamesDrawerBadgeRequest, ShowGamesDrawerBadgeResponse, } from "./gamesdrawerbadge_msg.js";
import { IsOptedInRequest, IsOptedInResponse, ListOptedInUsersRequest, ListOptedInUsersResponse, OptInCurrentUserResponse, OptOutCurrentUserResponse, } from "./optin_msg.js";
import { EnqueueRequest, EnqueueResponse } from "./pushnotif_msg.js";
export const NotificationsServiceName = "devvit.plugin.notifications.Notifications";
export class NotificationsClientImpl {
    constructor(rpc, opts) {
        this.service = opts?.service || NotificationsServiceName;
        this.rpc = rpc;
        this.OptInCurrentUser = this.OptInCurrentUser.bind(this);
        this.OptOutCurrentUser = this.OptOutCurrentUser.bind(this);
        this.ListOptedInUsers = this.ListOptedInUsers.bind(this);
        this.IsOptedIn = this.IsOptedIn.bind(this);
        this.Enqueue = this.Enqueue.bind(this);
        this.ShowGamesDrawerBadge = this.ShowGamesDrawerBadge.bind(this);
        this.DismissGamesDrawerBadge = this.DismissGamesDrawerBadge.bind(this);
        this.GetGamesDrawerBadgeStatus = this.GetGamesDrawerBadgeStatus.bind(this);
    }
    OptInCurrentUser(request, metadata) {
        const data = Empty.encode(request).finish();
        const promise = this.rpc.request(this.service, "OptInCurrentUser", data, metadata);
        return promise.then((data) => OptInCurrentUserResponse.decode(_m0.Reader.create(data)));
    }
    OptOutCurrentUser(request, metadata) {
        const data = Empty.encode(request).finish();
        const promise = this.rpc.request(this.service, "OptOutCurrentUser", data, metadata);
        return promise.then((data) => OptOutCurrentUserResponse.decode(_m0.Reader.create(data)));
    }
    ListOptedInUsers(request, metadata) {
        const data = ListOptedInUsersRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "ListOptedInUsers", data, metadata);
        return promise.then((data) => ListOptedInUsersResponse.decode(_m0.Reader.create(data)));
    }
    IsOptedIn(request, metadata) {
        const data = IsOptedInRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "IsOptedIn", data, metadata);
        return promise.then((data) => IsOptedInResponse.decode(_m0.Reader.create(data)));
    }
    Enqueue(request, metadata) {
        const data = EnqueueRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "Enqueue", data, metadata);
        return promise.then((data) => EnqueueResponse.decode(_m0.Reader.create(data)));
    }
    ShowGamesDrawerBadge(request, metadata) {
        const data = ShowGamesDrawerBadgeRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "ShowGamesDrawerBadge", data, metadata);
        return promise.then((data) => ShowGamesDrawerBadgeResponse.decode(_m0.Reader.create(data)));
    }
    DismissGamesDrawerBadge(request, metadata) {
        const data = DismissGamesDrawerBadgeRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "DismissGamesDrawerBadge", data, metadata);
        return promise.then((data) => DismissGamesDrawerBadgeResponse.decode(_m0.Reader.create(data)));
    }
    GetGamesDrawerBadgeStatus(request, metadata) {
        const data = GetGamesDrawerBadgeStatusRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "GetGamesDrawerBadgeStatus", data, metadata);
        return promise.then((data) => GetGamesDrawerBadgeStatusResponse.decode(_m0.Reader.create(data)));
    }
}
export const NotificationsDefinition = {
    name: "Notifications",
    fullName: "devvit.plugin.notifications.Notifications",
    methods: {
        /** This method opts the current authenticated user into receiving notifications. */
        optInCurrentUser: {
            name: "OptInCurrentUser",
            requestType: Empty,
            requestStream: false,
            responseType: OptInCurrentUserResponse,
            responseStream: false,
            options: {},
        },
        /** This method opts the current authenticated user out of receiving notifications. */
        optOutCurrentUser: {
            name: "OptOutCurrentUser",
            requestType: Empty,
            requestStream: false,
            responseType: OptOutCurrentUserResponse,
            responseStream: false,
            options: {},
        },
        /** Lists users who have opted in to receive notifications. */
        listOptedInUsers: {
            name: "ListOptedInUsers",
            requestType: ListOptedInUsersRequest,
            requestStream: false,
            responseType: ListOptedInUsersResponse,
            responseStream: false,
            options: {},
        },
        /** Checks if a specific user has opted in to receive notifications. */
        isOptedIn: {
            name: "IsOptedIn",
            requestType: IsOptedInRequest,
            requestStream: false,
            responseType: IsOptedInResponse,
            responseStream: false,
            options: {},
        },
        /**
         * This method queues a notification for asynchronous delivery. The actual
         * delivery timing and success is not guaranteed by this service.
         */
        enqueue: {
            name: "Enqueue",
            requestType: EnqueueRequest,
            requestStream: false,
            responseType: EnqueueResponse,
            responseStream: false,
            options: {},
        },
        /** Show a badge in the games drawer for a given app, linking to a given post */
        showGamesDrawerBadge: {
            name: "ShowGamesDrawerBadge",
            requestType: ShowGamesDrawerBadgeRequest,
            requestStream: false,
            responseType: ShowGamesDrawerBadgeResponse,
            responseStream: false,
            options: {},
        },
        /** Dismiss all active games drawer badges */
        dismissGamesDrawerBadge: {
            name: "DismissGamesDrawerBadge",
            requestType: DismissGamesDrawerBadgeRequest,
            requestStream: false,
            responseType: DismissGamesDrawerBadgeResponse,
            responseStream: false,
            options: {},
        },
        /** Get the status of active games drawer badges for an app */
        getGamesDrawerBadgeStatus: {
            name: "GetGamesDrawerBadgeStatus",
            requestType: GetGamesDrawerBadgeStatusRequest,
            requestStream: false,
            responseType: GetGamesDrawerBadgeStatusResponse,
            responseStream: false,
            options: {},
        },
    },
};
