// @devvit/protos/types/devvit/plugin/notifications/notifications_internal_svc.twirp.js
import { TwirpServer, TwirpError, TwirpErrorCode, TwirpContentType, chainInterceptors, } from 'twirp-ts';
import { GetGamesDrawerBadgeStatusForInstallationsRequest, GetGamesDrawerBadgeStatusForInstallationsResponse, } from './gamesdrawerbadge_msg.js';
export var NotificationsInternalMethod;
(function (NotificationsInternalMethod) {
    NotificationsInternalMethod["GetGamesDrawerBadgeStatusForInstallations"] = "GetGamesDrawerBadgeStatusForInstallations";
})(NotificationsInternalMethod || (NotificationsInternalMethod = {}));
export const NotificationsInternalMethodList = [
    NotificationsInternalMethod.GetGamesDrawerBadgeStatusForInstallations,
];
export function createNotificationsInternalServer(service) {
    return new TwirpServer({
        service,
        packageName: 'devvit.plugin.notifications',
        serviceName: 'NotificationsInternal',
        methodList: NotificationsInternalMethodList,
        matchRoute: matchNotificationsInternalRoute,
    });
}
function matchNotificationsInternalRoute(method, events) {
    switch (method) {
        case 'GetGamesDrawerBadgeStatusForInstallations':
            return async (ctx, service, data, interceptors) => {
                ctx = {
                    ...ctx,
                    methodName: 'GetGamesDrawerBadgeStatusForInstallations',
                };
                await events.onMatch(ctx);
                return handleNotificationsInternalGetGamesDrawerBadgeStatusForInstallationsRequest(ctx, service, data, interceptors);
            };
        default:
            events.onNotFound();
            const msg = `no handler found`;
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
function handleNotificationsInternalGetGamesDrawerBadgeStatusForInstallationsRequest(ctx, service, data, interceptors) {
    switch (ctx.contentType) {
        case TwirpContentType.JSON:
            return handleNotificationsInternalGetGamesDrawerBadgeStatusForInstallationsJSON(ctx, service, data, interceptors);
        case TwirpContentType.Protobuf:
            return handleNotificationsInternalGetGamesDrawerBadgeStatusForInstallationsProtobuf(ctx, service, data, interceptors);
        default:
            const msg = 'unexpected Content-Type';
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
async function handleNotificationsInternalGetGamesDrawerBadgeStatusForInstallationsJSON(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        const body = JSON.parse(data.toString() || '{}');
        request = GetGamesDrawerBadgeStatusForInstallationsRequest.fromJSON(body);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the json request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.GetGamesDrawerBadgeStatusForInstallations(ctx, inputReq);
        });
    }
    else {
        response = await service.GetGamesDrawerBadgeStatusForInstallations(ctx, request);
    }
    return JSON.stringify(GetGamesDrawerBadgeStatusForInstallationsResponse.toJSON(response));
}
async function handleNotificationsInternalGetGamesDrawerBadgeStatusForInstallationsProtobuf(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        request = GetGamesDrawerBadgeStatusForInstallationsRequest.decode(data);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the protobuf request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.GetGamesDrawerBadgeStatusForInstallations(ctx, inputReq);
        });
    }
    else {
        response = await service.GetGamesDrawerBadgeStatusForInstallations(ctx, request);
    }
    return Buffer.from(GetGamesDrawerBadgeStatusForInstallationsResponse.encode(response).finish());
}
