// @devvit/protos/types/devvit/plugin/notifications/notifications_internal_svc.js
/**
 * #notifications_internal_svc.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
import _m0 from "protobufjs/minimal.js";
import { GetGamesDrawerBadgeStatusForInstallationsRequest, GetGamesDrawerBadgeStatusForInstallationsResponse, } from "./gamesdrawerbadge_msg.js";
export const NotificationsInternalServiceName = "devvit.plugin.notifications.NotificationsInternal";
export class NotificationsInternalClientImpl {
    constructor(rpc, opts) {
        this.service = opts?.service || NotificationsInternalServiceName;
        this.rpc = rpc;
        this.GetGamesDrawerBadgeStatusForInstallations = this.GetGamesDrawerBadgeStatusForInstallations.bind(this);
    }
    GetGamesDrawerBadgeStatusForInstallations(request, metadata) {
        const data = GetGamesDrawerBadgeStatusForInstallationsRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "GetGamesDrawerBadgeStatusForInstallations", data, metadata);
        return promise.then((data) => GetGamesDrawerBadgeStatusForInstallationsResponse.decode(_m0.Reader.create(data)));
    }
}
export const NotificationsInternalDefinition = {
    name: "NotificationsInternal",
    fullName: "devvit.plugin.notifications.NotificationsInternal",
    methods: {
        /** Get the status of games drawer badges for installations specified by ID */
        getGamesDrawerBadgeStatusForInstallations: {
            name: "GetGamesDrawerBadgeStatusForInstallations",
            requestType: GetGamesDrawerBadgeStatusForInstallationsRequest,
            requestStream: false,
            responseType: GetGamesDrawerBadgeStatusForInstallationsResponse,
            responseStream: false,
            options: {},
        },
    },
};
