// @devvit/protos/types/devvit/plugin/notifications/optin_msg.js
/**
 * #optin_msg.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
import Long from "long";
import _m0 from "protobufjs/minimal.js";
import { messageTypeRegistry } from "../../../typeRegistry.js";
function createBaseOptInCurrentUserResponse() {
    return { success: false, message: "" };
}
export const OptInCurrentUserResponse = {
    $type: "devvit.plugin.notifications.OptInCurrentUserResponse",
    encode(message, writer = _m0.Writer.create()) {
        if (message.success !== false) {
            writer.uint32(8).bool(message.success);
        }
        if (message.message !== "") {
            writer.uint32(18).string(message.message);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseOptInCurrentUserResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.success = reader.bool();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.message = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            success: isSet(object.success) ? globalThis.Boolean(object.success) : false,
            message: isSet(object.message) ? globalThis.String(object.message) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.success !== false) {
            obj.success = message.success;
        }
        if (message.message !== "") {
            obj.message = message.message;
        }
        return obj;
    },
    create(base) {
        return OptInCurrentUserResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseOptInCurrentUserResponse();
        message.success = object.success ?? false;
        message.message = object.message ?? "";
        return message;
    },
};
messageTypeRegistry.set(OptInCurrentUserResponse.$type, OptInCurrentUserResponse);
function createBaseOptOutCurrentUserResponse() {
    return { success: false, message: "" };
}
export const OptOutCurrentUserResponse = {
    $type: "devvit.plugin.notifications.OptOutCurrentUserResponse",
    encode(message, writer = _m0.Writer.create()) {
        if (message.success !== false) {
            writer.uint32(8).bool(message.success);
        }
        if (message.message !== "") {
            writer.uint32(18).string(message.message);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseOptOutCurrentUserResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.success = reader.bool();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.message = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            success: isSet(object.success) ? globalThis.Boolean(object.success) : false,
            message: isSet(object.message) ? globalThis.String(object.message) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.success !== false) {
            obj.success = message.success;
        }
        if (message.message !== "") {
            obj.message = message.message;
        }
        return obj;
    },
    create(base) {
        return OptOutCurrentUserResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseOptOutCurrentUserResponse();
        message.success = object.success ?? false;
        message.message = object.message ?? "";
        return message;
    },
};
messageTypeRegistry.set(OptOutCurrentUserResponse.$type, OptOutCurrentUserResponse);
function createBaseListOptedInUsersRequest() {
    return { limit: undefined, after: undefined };
}
export const ListOptedInUsersRequest = {
    $type: "devvit.plugin.notifications.ListOptedInUsersRequest",
    encode(message, writer = _m0.Writer.create()) {
        if (message.limit !== undefined) {
            writer.uint32(8).int64(message.limit);
        }
        if (message.after !== undefined) {
            writer.uint32(18).string(message.after);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseListOptedInUsersRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.limit = longToNumber(reader.int64());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.after = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            limit: isSet(object.limit) ? globalThis.Number(object.limit) : undefined,
            after: isSet(object.after) ? globalThis.String(object.after) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.limit !== undefined) {
            obj.limit = Math.round(message.limit);
        }
        if (message.after !== undefined) {
            obj.after = message.after;
        }
        return obj;
    },
    create(base) {
        return ListOptedInUsersRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseListOptedInUsersRequest();
        message.limit = object.limit ?? undefined;
        message.after = object.after ?? undefined;
        return message;
    },
};
messageTypeRegistry.set(ListOptedInUsersRequest.$type, ListOptedInUsersRequest);
function createBaseListOptedInUsersResponse() {
    return { userIds: [], next: undefined };
}
export const ListOptedInUsersResponse = {
    $type: "devvit.plugin.notifications.ListOptedInUsersResponse",
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.userIds) {
            writer.uint32(10).string(v);
        }
        if (message.next !== undefined) {
            writer.uint32(18).string(message.next);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseListOptedInUsersResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.userIds.push(reader.string());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.next = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            userIds: globalThis.Array.isArray(object?.userIds) ? object.userIds.map((e) => globalThis.String(e)) : [],
            next: isSet(object.next) ? globalThis.String(object.next) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.userIds?.length) {
            obj.userIds = message.userIds;
        }
        if (message.next !== undefined) {
            obj.next = message.next;
        }
        return obj;
    },
    create(base) {
        return ListOptedInUsersResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseListOptedInUsersResponse();
        message.userIds = object.userIds?.map((e) => e) || [];
        message.next = object.next ?? undefined;
        return message;
    },
};
messageTypeRegistry.set(ListOptedInUsersResponse.$type, ListOptedInUsersResponse);
function createBaseIsOptedInRequest() {
    return { userId: "" };
}
export const IsOptedInRequest = {
    $type: "devvit.plugin.notifications.IsOptedInRequest",
    encode(message, writer = _m0.Writer.create()) {
        if (message.userId !== "") {
            writer.uint32(10).string(message.userId);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseIsOptedInRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.userId = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { userId: isSet(object.userId) ? globalThis.String(object.userId) : "" };
    },
    toJSON(message) {
        const obj = {};
        if (message.userId !== "") {
            obj.userId = message.userId;
        }
        return obj;
    },
    create(base) {
        return IsOptedInRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseIsOptedInRequest();
        message.userId = object.userId ?? "";
        return message;
    },
};
messageTypeRegistry.set(IsOptedInRequest.$type, IsOptedInRequest);
function createBaseIsOptedInResponse() {
    return { optedIn: false };
}
export const IsOptedInResponse = {
    $type: "devvit.plugin.notifications.IsOptedInResponse",
    encode(message, writer = _m0.Writer.create()) {
        if (message.optedIn !== false) {
            writer.uint32(8).bool(message.optedIn);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseIsOptedInResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.optedIn = reader.bool();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { optedIn: isSet(object.optedIn) ? globalThis.Boolean(object.optedIn) : false };
    },
    toJSON(message) {
        const obj = {};
        if (message.optedIn !== false) {
            obj.optedIn = message.optedIn;
        }
        return obj;
    },
    create(base) {
        return IsOptedInResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseIsOptedInResponse();
        message.optedIn = object.optedIn ?? false;
        return message;
    },
};
messageTypeRegistry.set(IsOptedInResponse.$type, IsOptedInResponse);
function longToNumber(long) {
    if (long.gt(globalThis.Number.MAX_SAFE_INTEGER)) {
        throw new globalThis.Error("Value is larger than Number.MAX_SAFE_INTEGER");
    }
    return long.toNumber();
}
if (_m0.util.Long !== Long) {
    _m0.util.Long = Long;
    _m0.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
