// @devvit/protos/types/devvit/plugin/notifications/pushnotif_msg.js
/**
 * #pushnotif_msg.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
import Long from "long";
import _m0 from "protobufjs/minimal.js";
import { Struct } from "../../../google/protobuf/struct.js";
import { messageTypeRegistry } from "../../../typeRegistry.js";
function createBaseEnqueueRequest() {
    return { title: "", body: "", recipients: [] };
}
export const EnqueueRequest = {
    $type: "devvit.plugin.notifications.EnqueueRequest",
    encode(message, writer = _m0.Writer.create()) {
        if (message.title !== "") {
            writer.uint32(10).string(message.title);
        }
        if (message.body !== "") {
            writer.uint32(18).string(message.body);
        }
        for (const v of message.recipients) {
            Recipient.encode(v, writer.uint32(42).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEnqueueRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.title = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.body = reader.string();
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    message.recipients.push(Recipient.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            title: isSet(object.title) ? globalThis.String(object.title) : "",
            body: isSet(object.body) ? globalThis.String(object.body) : "",
            recipients: globalThis.Array.isArray(object?.recipients)
                ? object.recipients.map((e) => Recipient.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.title !== "") {
            obj.title = message.title;
        }
        if (message.body !== "") {
            obj.body = message.body;
        }
        if (message.recipients?.length) {
            obj.recipients = message.recipients.map((e) => Recipient.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return EnqueueRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseEnqueueRequest();
        message.title = object.title ?? "";
        message.body = object.body ?? "";
        message.recipients = object.recipients?.map((e) => Recipient.fromPartial(e)) || [];
        return message;
    },
};
messageTypeRegistry.set(EnqueueRequest.$type, EnqueueRequest);
function createBaseEnqueueResponse() {
    return { successCount: 0, failureCount: 0, errors: [], timestamp: 0 };
}
export const EnqueueResponse = {
    $type: "devvit.plugin.notifications.EnqueueResponse",
    encode(message, writer = _m0.Writer.create()) {
        if (message.successCount !== 0) {
            writer.uint32(8).int32(message.successCount);
        }
        if (message.failureCount !== 0) {
            writer.uint32(16).int32(message.failureCount);
        }
        for (const v of message.errors) {
            EnqueueError.encode(v, writer.uint32(26).fork()).ldelim();
        }
        if (message.timestamp !== 0) {
            writer.uint32(32).int64(message.timestamp);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEnqueueResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.successCount = reader.int32();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.failureCount = reader.int32();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.errors.push(EnqueueError.decode(reader, reader.uint32()));
                    continue;
                case 4:
                    if (tag !== 32) {
                        break;
                    }
                    message.timestamp = longToNumber(reader.int64());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            successCount: isSet(object.successCount) ? globalThis.Number(object.successCount) : 0,
            failureCount: isSet(object.failureCount) ? globalThis.Number(object.failureCount) : 0,
            errors: globalThis.Array.isArray(object?.errors) ? object.errors.map((e) => EnqueueError.fromJSON(e)) : [],
            timestamp: isSet(object.timestamp) ? globalThis.Number(object.timestamp) : 0,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.successCount !== 0) {
            obj.successCount = Math.round(message.successCount);
        }
        if (message.failureCount !== 0) {
            obj.failureCount = Math.round(message.failureCount);
        }
        if (message.errors?.length) {
            obj.errors = message.errors.map((e) => EnqueueError.toJSON(e));
        }
        if (message.timestamp !== 0) {
            obj.timestamp = Math.round(message.timestamp);
        }
        return obj;
    },
    create(base) {
        return EnqueueResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseEnqueueResponse();
        message.successCount = object.successCount ?? 0;
        message.failureCount = object.failureCount ?? 0;
        message.errors = object.errors?.map((e) => EnqueueError.fromPartial(e)) || [];
        message.timestamp = object.timestamp ?? 0;
        return message;
    },
};
messageTypeRegistry.set(EnqueueResponse.$type, EnqueueResponse);
function createBaseRecipient() {
    return { userId: "", comment: undefined, post: undefined, data: undefined };
}
export const Recipient = {
    $type: "devvit.plugin.notifications.Recipient",
    encode(message, writer = _m0.Writer.create()) {
        if (message.userId !== "") {
            writer.uint32(10).string(message.userId);
        }
        if (message.comment !== undefined) {
            writer.uint32(18).string(message.comment);
        }
        if (message.post !== undefined) {
            writer.uint32(26).string(message.post);
        }
        if (message.data !== undefined) {
            Struct.encode(Struct.wrap(message.data), writer.uint32(34).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRecipient();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.userId = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.comment = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.post = reader.string();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.data = Struct.unwrap(Struct.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            userId: isSet(object.userId) ? globalThis.String(object.userId) : "",
            comment: isSet(object.comment) ? globalThis.String(object.comment) : undefined,
            post: isSet(object.post) ? globalThis.String(object.post) : undefined,
            data: isObject(object.data) ? object.data : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.userId !== "") {
            obj.userId = message.userId;
        }
        if (message.comment !== undefined) {
            obj.comment = message.comment;
        }
        if (message.post !== undefined) {
            obj.post = message.post;
        }
        if (message.data !== undefined) {
            obj.data = message.data;
        }
        return obj;
    },
    create(base) {
        return Recipient.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseRecipient();
        message.userId = object.userId ?? "";
        message.comment = object.comment ?? undefined;
        message.post = object.post ?? undefined;
        message.data = object.data ?? undefined;
        return message;
    },
};
messageTypeRegistry.set(Recipient.$type, Recipient);
function createBaseEnqueueError() {
    return { userId: "", message: "" };
}
export const EnqueueError = {
    $type: "devvit.plugin.notifications.EnqueueError",
    encode(message, writer = _m0.Writer.create()) {
        if (message.userId !== "") {
            writer.uint32(10).string(message.userId);
        }
        if (message.message !== "") {
            writer.uint32(18).string(message.message);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEnqueueError();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.userId = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.message = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            userId: isSet(object.userId) ? globalThis.String(object.userId) : "",
            message: isSet(object.message) ? globalThis.String(object.message) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.userId !== "") {
            obj.userId = message.userId;
        }
        if (message.message !== "") {
            obj.message = message.message;
        }
        return obj;
    },
    create(base) {
        return EnqueueError.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseEnqueueError();
        message.userId = object.userId ?? "";
        message.message = object.message ?? "";
        return message;
    },
};
messageTypeRegistry.set(EnqueueError.$type, EnqueueError);
function longToNumber(long) {
    if (long.gt(globalThis.Number.MAX_SAFE_INTEGER)) {
        throw new globalThis.Error("Value is larger than Number.MAX_SAFE_INTEGER");
    }
    return long.toNumber();
}
if (_m0.util.Long !== Long) {
    _m0.util.Long = Long;
    _m0.configure();
}
function isObject(value) {
    return typeof value === "object" && value !== null;
}
function isSet(value) {
    return value !== null && value !== undefined;
}
