// @devvit/tsconfig/base-vitest.config.js
import { defineConfig } from 'vitest/config';
/** @import { ViteUserConfig, UserWorkspaceConfig } from 'vitest/config' */

/** @type {ViteUserConfig['test']} */
const baseConfig = {
  // Enables import-less tests for drop-in jest compatibility
  globals: true,
  pool: 'threads',
  environment: 'node',
  exclude: ['dist', 'node_modules'],
  coverage: {
    exclude: ['dist', 'node_modules'],
  },
  clearMocks: true,
  restoreMocks: true,
  // Automatically retry any test failures once.
  // Ideally we could scope this down to timeout failures only, but that doesn't seem currently possible.
  // This causes all snapshot tests to pass since it produces a new snapshot on the retry. Turning off until this is resolved
  // https://github.com/vitest-dev/vitest/issues/5312
  // retry: 1,
};

const DOM_TEST_FILE_PATTERN = '**/*.ui.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}';

export const devvitVitestConfig = /** @type {UserWorkspaceConfig} */ (
  defineConfig({
    test: { ...baseConfig, exclude: [...(baseConfig.exclude ?? []), DOM_TEST_FILE_PATTERN] },
  })
);

export const devvitVitestDomConfig = /** @type {UserWorkspaceConfig} */ (
  defineConfig({
    test: { ...baseConfig, environment: 'jsdom', include: [DOM_TEST_FILE_PATTERN] },
  })
);
