// @devvit/protos/json/devvit/ui/form_builder/v1alpha/type.js
/**
 * #type.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
export var FormFieldType;
(function (FormFieldType) {
    /** STRING - Single-line string */
    FormFieldType[FormFieldType["STRING"] = 0] = "STRING";
    /** PARAGRAPH - Multi-line string */
    FormFieldType[FormFieldType["PARAGRAPH"] = 1] = "PARAGRAPH";
    /** NUMBER - Number input */
    FormFieldType[FormFieldType["NUMBER"] = 2] = "NUMBER";
    /** BOOLEAN - Checkbox, switch */
    FormFieldType[FormFieldType["BOOLEAN"] = 3] = "BOOLEAN";
    /** LIST - Collect multiple entries of a single type */
    FormFieldType[FormFieldType["LIST"] = 4] = "LIST";
    /** SELECTION - Provide a list of options for the user to choose from */
    FormFieldType[FormFieldType["SELECTION"] = 5] = "SELECTION";
    /** GROUP - Create a labeled subsection of the form with nested fields */
    FormFieldType[FormFieldType["GROUP"] = 6] = "GROUP";
    /** IMAGE - Image upload - value is a URL for the uploaded image */
    FormFieldType[FormFieldType["IMAGE"] = 7] = "IMAGE";
    FormFieldType[FormFieldType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(FormFieldType || (FormFieldType = {}));
