// @devvit/shared-types/test/index.js
import '../shared/devvit-worker-global.js';
import { Header } from '../Header.js';
export const MOCK_HEADERS = {
    [Header.Subreddit]: 't5_testsub',
    [Header.SubredditName]: 'testsub',
    [Header.User]: 't2_testuser',
    [Header.AppUser]: 't2_testuser',
    [Header.App]: 'test-app',
    [Header.Version]: '0.0.0-test',
    [Header.AppViewerAuthToken]: 'test-token',
};
export const installGlobalConfig = (config, appConfig) => {
    var _a;
    globalThis.devvit ?? (globalThis.devvit = {});
    globalThis.devvit.config = config;
    (_a = globalThis.devvit).compute ?? (_a.compute = { platform: 'node' });
    globalThis.devvit.appConfig = appConfig ?? getDefaultAppConfig();
};
export const getDefaultAppConfig = () => {
    return {
        schema: 'v1',
        name: 'test-app',
        permissions: {
            http: { enable: true, domains: [] },
            media: true,
            menu: true,
            payments: true,
            realtime: true,
            redis: true,
            reddit: { enable: true, scope: 'user', asUser: [] },
            settings: true,
            triggers: true,
        },
        post: {
            dir: 'webroot',
            entrypoints: {
                default: {
                    inline: true,
                    entry: 'splash.html',
                    height: 'tall',
                    name: 'default',
                },
                game: {
                    entry: 'game.html',
                    height: 'regular',
                    name: 'game',
                },
            },
        },
        json: {
            name: 'test-app',
            permissions: { reddit: true },
        },
    };
};
export const makeConfig = ({ plugins = {}, } = {}) => {
    return {
        assets: {},
        providedDefinitions: [],
        webviewAssets: {},
        getPermissions: () => [],
        export: () => ({
            hostname: '',
            provides: [],
            uses: [],
            permissions: [],
        }),
        provides: () => { },
        addPermissions: () => { },
        use(definition) {
            if (definition.fullName in plugins) {
                return plugins[definition.fullName];
            }
            throw new Error(`Plugin not mocked: ${definition.fullName}`);
        },
        uses(definition) {
            return definition.fullName in plugins;
        },
    };
};
