// @devvit/protos/json/devvit/options/options.js
/**
 * #options.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
/**
 * Data Tier defines the level of data delivered to the client
 * FIRST_DATA_TIER is the least amount of data, THIRD_DATA_TIER is the most
 */
export var DataTier;
(function (DataTier) {
    DataTier[DataTier["UNKNOWN_DATA_TIER"] = 0] = "UNKNOWN_DATA_TIER";
    DataTier[DataTier["FIRST_DATA_TIER"] = 1] = "FIRST_DATA_TIER";
    DataTier[DataTier["SECOND_DATA_TIER"] = 2] = "SECOND_DATA_TIER";
    DataTier[DataTier["THIRD_DATA_TIER"] = 3] = "THIRD_DATA_TIER";
    DataTier[DataTier["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(DataTier || (DataTier = {}));
export var Selector;
(function (Selector) {
    /** @deprecated */
    Selector[Selector["DEFAULT"] = 0] = "DEFAULT";
    /** @deprecated */
    Selector[Selector["TARGET_USER"] = 1] = "TARGET_USER";
    /**
     * MOD_ACTION_TYPE - TODO(logan.hanks): Replace this with a field path string
     *
     * @deprecated
     */
    Selector[Selector["MOD_ACTION_TYPE"] = 2] = "MOD_ACTION_TYPE";
    Selector[Selector["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(Selector || (Selector = {}));
export var DeliveryLocation;
(function (DeliveryLocation) {
    DeliveryLocation[DeliveryLocation["UNKNOWN_DELIVERY_LOCATION"] = 0] = "UNKNOWN_DELIVERY_LOCATION";
    DeliveryLocation[DeliveryLocation["GATEWAY"] = 1] = "GATEWAY";
    DeliveryLocation[DeliveryLocation["FIREHOSE"] = 2] = "FIREHOSE";
    DeliveryLocation[DeliveryLocation["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(DeliveryLocation || (DeliveryLocation = {}));
/**
 * Event types for firehose
 * Also maps directly to the stream name in redis
 */
export var FirehoseEvent;
(function (FirehoseEvent) {
    FirehoseEvent[FirehoseEvent["UNKNOWN_FIREHOSE_EVENT"] = 0] = "UNKNOWN_FIREHOSE_EVENT";
    FirehoseEvent[FirehoseEvent["POST_CREATE"] = 1] = "POST_CREATE";
    FirehoseEvent[FirehoseEvent["POST_EDIT"] = 2] = "POST_EDIT";
    FirehoseEvent[FirehoseEvent["POST_DELETE"] = 3] = "POST_DELETE";
    FirehoseEvent[FirehoseEvent["COMMENT_CREATE"] = 4] = "COMMENT_CREATE";
    FirehoseEvent[FirehoseEvent["COMMENT_EDIT"] = 5] = "COMMENT_EDIT";
    FirehoseEvent[FirehoseEvent["COMMENT_DELETE"] = 6] = "COMMENT_DELETE";
    FirehoseEvent[FirehoseEvent["VOTE"] = 7] = "VOTE";
    FirehoseEvent[FirehoseEvent["MOD_ACTION"] = 8] = "MOD_ACTION";
    FirehoseEvent[FirehoseEvent["ACCOUNT_DELETE"] = 9] = "ACCOUNT_DELETE";
    FirehoseEvent[FirehoseEvent["ERROR"] = 10] = "ERROR";
    FirehoseEvent[FirehoseEvent["WARNING"] = 11] = "WARNING";
    FirehoseEvent[FirehoseEvent["POST_APPROVE"] = 12] = "POST_APPROVE";
    FirehoseEvent[FirehoseEvent["COMMENT_APPROVE"] = 13] = "COMMENT_APPROVE";
    FirehoseEvent[FirehoseEvent["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(FirehoseEvent || (FirehoseEvent = {}));
