// @devvit/protos/json/devvit/payments/v1alpha/product.js
/**
 * #product.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
export var Currency;
(function (Currency) {
    Currency[Currency["CURRENCY_UNSPECIFIED"] = 0] = "CURRENCY_UNSPECIFIED";
    /** GOLD - Reddit internal currency */
    Currency[Currency["GOLD"] = 200] = "GOLD";
    Currency[Currency["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(Currency || (Currency = {}));
export var AccountingType;
(function (AccountingType) {
    AccountingType[AccountingType["ACCOUNTING_TYPE_UNSPECIFIED"] = 0] = "ACCOUNTING_TYPE_UNSPECIFIED";
    /** ACCOUNTING_TYPE_INSTANT - Used immediately as it is purchased. */
    AccountingType[AccountingType["ACCOUNTING_TYPE_INSTANT"] = 1] = "ACCOUNTING_TYPE_INSTANT";
    /** ACCOUNTING_TYPE_DURABLE - It grants access indefinitely. */
    AccountingType[AccountingType["ACCOUNTING_TYPE_DURABLE"] = 2] = "ACCOUNTING_TYPE_DURABLE";
    /** ACCOUNTING_TYPE_CONSUMABLE - User can use the good at any time in the future but it is gone once it's used. */
    AccountingType[AccountingType["ACCOUNTING_TYPE_CONSUMABLE"] = 3] = "ACCOUNTING_TYPE_CONSUMABLE";
    /** ACCOUNTING_TYPE_VALID_FOR_1D - Consumed over a fixed period of time. */
    AccountingType[AccountingType["ACCOUNTING_TYPE_VALID_FOR_1D"] = 4] = "ACCOUNTING_TYPE_VALID_FOR_1D";
    AccountingType[AccountingType["ACCOUNTING_TYPE_VALID_FOR_7D"] = 5] = "ACCOUNTING_TYPE_VALID_FOR_7D";
    AccountingType[AccountingType["ACCOUNTING_TYPE_VALID_FOR_30D"] = 6] = "ACCOUNTING_TYPE_VALID_FOR_30D";
    AccountingType[AccountingType["ACCOUNTING_TYPE_VALID_FOR_1Y"] = 7] = "ACCOUNTING_TYPE_VALID_FOR_1Y";
    AccountingType[AccountingType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(AccountingType || (AccountingType = {}));
