// @devvit/protos/types/devvit/gateway/app_info/v1alpha/app_info.js
/**
 * #app_info.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
import _m0 from "protobufjs/minimal.js";
import { Timestamp } from "../../../../google/protobuf/timestamp.js";
import { messageTypeRegistry } from "../../../../typeRegistry.js";
function createBaseGetAppInfoBySlugRequest() {
    return { slug: "" };
}
export const GetAppInfoBySlugRequest = {
    $type: "devvit.gateway.app_info.v1alpha.GetAppInfoBySlugRequest",
    encode(message, writer = _m0.Writer.create()) {
        if (message.slug !== "") {
            writer.uint32(10).string(message.slug);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetAppInfoBySlugRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.slug = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { slug: isSet(object.slug) ? globalThis.String(object.slug) : "" };
    },
    toJSON(message) {
        const obj = {};
        if (message.slug !== "") {
            obj.slug = message.slug;
        }
        return obj;
    },
    create(base) {
        return GetAppInfoBySlugRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetAppInfoBySlugRequest();
        message.slug = object.slug ?? "";
        return message;
    },
};
messageTypeRegistry.set(GetAppInfoBySlugRequest.$type, GetAppInfoBySlugRequest);
function createBaseGetAppInfoBySlugResponse() {
    return { app: undefined };
}
export const GetAppInfoBySlugResponse = {
    $type: "devvit.gateway.app_info.v1alpha.GetAppInfoBySlugResponse",
    encode(message, writer = _m0.Writer.create()) {
        if (message.app !== undefined) {
            AppInfo.encode(message.app, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetAppInfoBySlugResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.app = AppInfo.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { app: isSet(object.app) ? AppInfo.fromJSON(object.app) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.app !== undefined) {
            obj.app = AppInfo.toJSON(message.app);
        }
        return obj;
    },
    create(base) {
        return GetAppInfoBySlugResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetAppInfoBySlugResponse();
        message.app = (object.app !== undefined && object.app !== null) ? AppInfo.fromPartial(object.app) : undefined;
        return message;
    },
};
messageTypeRegistry.set(GetAppInfoBySlugResponse.$type, GetAppInfoBySlugResponse);
function createBaseAppInfo() {
    return {
        id: "",
        appAccountId: undefined,
        slugBase: "",
        slug: "",
        name: "",
        description: "",
        isNsfw: false,
        isDelisted: false,
        isArchived: false,
        isFirstParty: false,
        isFlagged: false,
        createdAt: undefined,
        ownerId: "",
        isDisabled: false,
    };
}
export const AppInfo = {
    $type: "devvit.gateway.app_info.v1alpha.AppInfo",
    encode(message, writer = _m0.Writer.create()) {
        if (message.id !== "") {
            writer.uint32(10).string(message.id);
        }
        if (message.appAccountId !== undefined) {
            writer.uint32(18).string(message.appAccountId);
        }
        if (message.slugBase !== "") {
            writer.uint32(26).string(message.slugBase);
        }
        if (message.slug !== "") {
            writer.uint32(34).string(message.slug);
        }
        if (message.name !== "") {
            writer.uint32(42).string(message.name);
        }
        if (message.description !== "") {
            writer.uint32(50).string(message.description);
        }
        if (message.isNsfw !== false) {
            writer.uint32(56).bool(message.isNsfw);
        }
        if (message.isDelisted !== false) {
            writer.uint32(64).bool(message.isDelisted);
        }
        if (message.isArchived !== false) {
            writer.uint32(72).bool(message.isArchived);
        }
        if (message.isFirstParty !== false) {
            writer.uint32(80).bool(message.isFirstParty);
        }
        if (message.isFlagged !== false) {
            writer.uint32(88).bool(message.isFlagged);
        }
        if (message.createdAt !== undefined) {
            Timestamp.encode(toTimestamp(message.createdAt), writer.uint32(98).fork()).ldelim();
        }
        if (message.ownerId !== "") {
            writer.uint32(106).string(message.ownerId);
        }
        if (message.isDisabled !== false) {
            writer.uint32(112).bool(message.isDisabled);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAppInfo();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.id = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.appAccountId = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.slugBase = reader.string();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.slug = reader.string();
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    message.name = reader.string();
                    continue;
                case 6:
                    if (tag !== 50) {
                        break;
                    }
                    message.description = reader.string();
                    continue;
                case 7:
                    if (tag !== 56) {
                        break;
                    }
                    message.isNsfw = reader.bool();
                    continue;
                case 8:
                    if (tag !== 64) {
                        break;
                    }
                    message.isDelisted = reader.bool();
                    continue;
                case 9:
                    if (tag !== 72) {
                        break;
                    }
                    message.isArchived = reader.bool();
                    continue;
                case 10:
                    if (tag !== 80) {
                        break;
                    }
                    message.isFirstParty = reader.bool();
                    continue;
                case 11:
                    if (tag !== 88) {
                        break;
                    }
                    message.isFlagged = reader.bool();
                    continue;
                case 12:
                    if (tag !== 98) {
                        break;
                    }
                    message.createdAt = fromTimestamp(Timestamp.decode(reader, reader.uint32()));
                    continue;
                case 13:
                    if (tag !== 106) {
                        break;
                    }
                    message.ownerId = reader.string();
                    continue;
                case 14:
                    if (tag !== 112) {
                        break;
                    }
                    message.isDisabled = reader.bool();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            id: isSet(object.id) ? globalThis.String(object.id) : "",
            appAccountId: isSet(object.appAccountId) ? globalThis.String(object.appAccountId) : undefined,
            slugBase: isSet(object.slugBase) ? globalThis.String(object.slugBase) : "",
            slug: isSet(object.slug) ? globalThis.String(object.slug) : "",
            name: isSet(object.name) ? globalThis.String(object.name) : "",
            description: isSet(object.description) ? globalThis.String(object.description) : "",
            isNsfw: isSet(object.isNsfw) ? globalThis.Boolean(object.isNsfw) : false,
            isDelisted: isSet(object.isDelisted) ? globalThis.Boolean(object.isDelisted) : false,
            isArchived: isSet(object.isArchived) ? globalThis.Boolean(object.isArchived) : false,
            isFirstParty: isSet(object.isFirstParty) ? globalThis.Boolean(object.isFirstParty) : false,
            isFlagged: isSet(object.isFlagged) ? globalThis.Boolean(object.isFlagged) : false,
            createdAt: isSet(object.createdAt) ? fromJsonTimestamp(object.createdAt) : undefined,
            ownerId: isSet(object.ownerId) ? globalThis.String(object.ownerId) : "",
            isDisabled: isSet(object.isDisabled) ? globalThis.Boolean(object.isDisabled) : false,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.id !== "") {
            obj.id = message.id;
        }
        if (message.appAccountId !== undefined) {
            obj.appAccountId = message.appAccountId;
        }
        if (message.slugBase !== "") {
            obj.slugBase = message.slugBase;
        }
        if (message.slug !== "") {
            obj.slug = message.slug;
        }
        if (message.name !== "") {
            obj.name = message.name;
        }
        if (message.description !== "") {
            obj.description = message.description;
        }
        if (message.isNsfw !== false) {
            obj.isNsfw = message.isNsfw;
        }
        if (message.isDelisted !== false) {
            obj.isDelisted = message.isDelisted;
        }
        if (message.isArchived !== false) {
            obj.isArchived = message.isArchived;
        }
        if (message.isFirstParty !== false) {
            obj.isFirstParty = message.isFirstParty;
        }
        if (message.isFlagged !== false) {
            obj.isFlagged = message.isFlagged;
        }
        if (message.createdAt !== undefined) {
            obj.createdAt = message.createdAt.toISOString();
        }
        if (message.ownerId !== "") {
            obj.ownerId = message.ownerId;
        }
        if (message.isDisabled !== false) {
            obj.isDisabled = message.isDisabled;
        }
        return obj;
    },
    create(base) {
        return AppInfo.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseAppInfo();
        message.id = object.id ?? "";
        message.appAccountId = object.appAccountId ?? undefined;
        message.slugBase = object.slugBase ?? "";
        message.slug = object.slug ?? "";
        message.name = object.name ?? "";
        message.description = object.description ?? "";
        message.isNsfw = object.isNsfw ?? false;
        message.isDelisted = object.isDelisted ?? false;
        message.isArchived = object.isArchived ?? false;
        message.isFirstParty = object.isFirstParty ?? false;
        message.isFlagged = object.isFlagged ?? false;
        message.createdAt = object.createdAt ?? undefined;
        message.ownerId = object.ownerId ?? "";
        message.isDisabled = object.isDisabled ?? false;
        return message;
    },
};
messageTypeRegistry.set(AppInfo.$type, AppInfo);
export const AppInfoServiceServiceName = "devvit.gateway.app_info.v1alpha.AppInfoService";
export class AppInfoServiceClientImpl {
    constructor(rpc, opts) {
        this.service = opts?.service || AppInfoServiceServiceName;
        this.rpc = rpc;
        this.GetAppInfoBySlug = this.GetAppInfoBySlug.bind(this);
    }
    GetAppInfoBySlug(request, metadata) {
        const data = GetAppInfoBySlugRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "GetAppInfoBySlug", data, metadata);
        return promise.then((data) => GetAppInfoBySlugResponse.decode(_m0.Reader.create(data)));
    }
}
export const AppInfoServiceDefinition = {
    name: "AppInfoService",
    fullName: "devvit.gateway.app_info.v1alpha.AppInfoService",
    methods: {
        /** GetAppInfoBySlug returns minimal information about an app identified by slug. */
        getAppInfoBySlug: {
            name: "GetAppInfoBySlug",
            requestType: GetAppInfoBySlugRequest,
            requestStream: false,
            responseType: GetAppInfoBySlugResponse,
            responseStream: false,
            options: {},
        },
    },
};
function toTimestamp(date) {
    const seconds = Math.trunc(date.getTime() / 1000);
    const nanos = (date.getTime() % 1000) * 1000000;
    return { seconds, nanos };
}
function fromTimestamp(t) {
    let millis = (t.seconds || 0) * 1000;
    millis += (t.nanos || 0) / 1000000;
    return new globalThis.Date(millis);
}
function fromJsonTimestamp(o) {
    if (o instanceof globalThis.Date) {
        return o;
    }
    else if (typeof o === "string") {
        return new globalThis.Date(o);
    }
    else {
        return fromTimestamp(Timestamp.fromJSON(o));
    }
}
function isSet(value) {
    return value !== null && value !== undefined;
}
