// @devvit/protos/types/devvit/gateway/app_info/v1alpha/app_info.twirp.js
import { TwirpServer, TwirpError, TwirpErrorCode, TwirpContentType, chainInterceptors, } from 'twirp-ts';
import { GetAppInfoBySlugRequest, GetAppInfoBySlugResponse } from './app_info.js';
export var AppInfoServiceMethod;
(function (AppInfoServiceMethod) {
    AppInfoServiceMethod["GetAppInfoBySlug"] = "GetAppInfoBySlug";
})(AppInfoServiceMethod || (AppInfoServiceMethod = {}));
export const AppInfoServiceMethodList = [AppInfoServiceMethod.GetAppInfoBySlug];
export function createAppInfoServiceServer(service) {
    return new TwirpServer({
        service,
        packageName: 'devvit.gateway.app_info.v1alpha',
        serviceName: 'AppInfoService',
        methodList: AppInfoServiceMethodList,
        matchRoute: matchAppInfoServiceRoute,
    });
}
function matchAppInfoServiceRoute(method, events) {
    switch (method) {
        case 'GetAppInfoBySlug':
            return async (ctx, service, data, interceptors) => {
                ctx = { ...ctx, methodName: 'GetAppInfoBySlug' };
                await events.onMatch(ctx);
                return handleAppInfoServiceGetAppInfoBySlugRequest(ctx, service, data, interceptors);
            };
        default:
            events.onNotFound();
            const msg = `no handler found`;
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
function handleAppInfoServiceGetAppInfoBySlugRequest(ctx, service, data, interceptors) {
    switch (ctx.contentType) {
        case TwirpContentType.JSON:
            return handleAppInfoServiceGetAppInfoBySlugJSON(ctx, service, data, interceptors);
        case TwirpContentType.Protobuf:
            return handleAppInfoServiceGetAppInfoBySlugProtobuf(ctx, service, data, interceptors);
        default:
            const msg = 'unexpected Content-Type';
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
async function handleAppInfoServiceGetAppInfoBySlugJSON(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        const body = JSON.parse(data.toString() || '{}');
        request = GetAppInfoBySlugRequest.fromJSON(body);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the json request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.GetAppInfoBySlug(ctx, inputReq);
        });
    }
    else {
        response = await service.GetAppInfoBySlug(ctx, request);
    }
    return JSON.stringify(GetAppInfoBySlugResponse.toJSON(response));
}
async function handleAppInfoServiceGetAppInfoBySlugProtobuf(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        request = GetAppInfoBySlugRequest.decode(data);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the protobuf request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.GetAppInfoBySlug(ctx, inputReq);
        });
    }
    else {
        response = await service.GetAppInfoBySlug(ctx, request);
    }
    return Buffer.from(GetAppInfoBySlugResponse.encode(response).finish());
}
