// @devvit/protos/types/devvit/ui/effects/web_view/v1alpha/reserved.js
/**
 * #reserved.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
import _m0 from "protobufjs/minimal.js";
import { messageTypeRegistry } from "../../../../../typeRegistry.js";
import { RequestContext } from "../../../../platform/v1/request_context.js";
function createBaseReserved() {
    return { requestContext: undefined };
}
export const Reserved = {
    $type: "devvit.ui.effects.web_view.v1alpha.Reserved",
    encode(message, writer = _m0.Writer.create()) {
        if (message.requestContext !== undefined) {
            RequestContext.encode(message.requestContext, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseReserved();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.requestContext = RequestContext.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            requestContext: isSet(object.requestContext) ? RequestContext.fromJSON(object.requestContext) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.requestContext !== undefined) {
            obj.requestContext = RequestContext.toJSON(message.requestContext);
        }
        return obj;
    },
    create(base) {
        return Reserved.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseReserved();
        message.requestContext = (object.requestContext !== undefined && object.requestContext !== null)
            ? RequestContext.fromPartial(object.requestContext)
            : undefined;
        return message;
    },
};
messageTypeRegistry.set(Reserved.$type, Reserved);
function isSet(value) {
    return value !== null && value !== undefined;
}
