// @devvit/protos/types/devvit/actor/webbit/webbit.twirp.js
import { TwirpServer, TwirpError, TwirpErrorCode, TwirpContentType, chainInterceptors, } from 'twirp-ts';
import { WebbitHttpRequest, WebbitHttpResponse } from './webbit.js';
export var WebbitServerMethod;
(function (WebbitServerMethod) {
    WebbitServerMethod["Request"] = "Request";
})(WebbitServerMethod || (WebbitServerMethod = {}));
export const WebbitServerMethodList = [WebbitServerMethod.Request];
export function createWebbitServerServer(service) {
    return new TwirpServer({
        service,
        packageName: 'devvit.actor.webbit',
        serviceName: 'WebbitServer',
        methodList: WebbitServerMethodList,
        matchRoute: matchWebbitServerRoute,
    });
}
function matchWebbitServerRoute(method, events) {
    switch (method) {
        case 'Request':
            return async (ctx, service, data, interceptors) => {
                ctx = { ...ctx, methodName: 'Request' };
                await events.onMatch(ctx);
                return handleWebbitServerRequestRequest(ctx, service, data, interceptors);
            };
        default:
            events.onNotFound();
            const msg = `no handler found`;
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
function handleWebbitServerRequestRequest(ctx, service, data, interceptors) {
    switch (ctx.contentType) {
        case TwirpContentType.JSON:
            return handleWebbitServerRequestJSON(ctx, service, data, interceptors);
        case TwirpContentType.Protobuf:
            return handleWebbitServerRequestProtobuf(ctx, service, data, interceptors);
        default:
            const msg = 'unexpected Content-Type';
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
async function handleWebbitServerRequestJSON(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        const body = JSON.parse(data.toString() || '{}');
        request = WebbitHttpRequest.fromJSON(body);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the json request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.Request(ctx, inputReq);
        });
    }
    else {
        response = await service.Request(ctx, request);
    }
    return JSON.stringify(WebbitHttpResponse.toJSON(response));
}
async function handleWebbitServerRequestProtobuf(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        request = WebbitHttpRequest.decode(data);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the protobuf request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.Request(ctx, inputReq);
        });
    }
    else {
        response = await service.Request(ctx, request);
    }
    return Buffer.from(WebbitHttpResponse.encode(response).finish());
}
