// @devvit/protos/types/devvit/gateway/playtest_subreddit/playtest_subreddit.twirp.js
import { TwirpServer, TwirpError, TwirpErrorCode, TwirpContentType, chainInterceptors, } from 'twirp-ts';
import { CreatePlaytestSubredditRequest, CreatePlaytestSubredditResponse, } from './playtest_subreddit.js';
export var PlaytestSubredditMethod;
(function (PlaytestSubredditMethod) {
    PlaytestSubredditMethod["CreatePlaytestSubreddit"] = "CreatePlaytestSubreddit";
})(PlaytestSubredditMethod || (PlaytestSubredditMethod = {}));
export const PlaytestSubredditMethodList = [
    PlaytestSubredditMethod.CreatePlaytestSubreddit,
];
export function createPlaytestSubredditServer(service) {
    return new TwirpServer({
        service,
        packageName: 'devvit.gateway.playtest_subreddit',
        serviceName: 'PlaytestSubreddit',
        methodList: PlaytestSubredditMethodList,
        matchRoute: matchPlaytestSubredditRoute,
    });
}
function matchPlaytestSubredditRoute(method, events) {
    switch (method) {
        case 'CreatePlaytestSubreddit':
            return async (ctx, service, data, interceptors) => {
                ctx = { ...ctx, methodName: 'CreatePlaytestSubreddit' };
                await events.onMatch(ctx);
                return handlePlaytestSubredditCreatePlaytestSubredditRequest(ctx, service, data, interceptors);
            };
        default:
            events.onNotFound();
            const msg = `no handler found`;
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
function handlePlaytestSubredditCreatePlaytestSubredditRequest(ctx, service, data, interceptors) {
    switch (ctx.contentType) {
        case TwirpContentType.JSON:
            return handlePlaytestSubredditCreatePlaytestSubredditJSON(ctx, service, data, interceptors);
        case TwirpContentType.Protobuf:
            return handlePlaytestSubredditCreatePlaytestSubredditProtobuf(ctx, service, data, interceptors);
        default:
            const msg = 'unexpected Content-Type';
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
async function handlePlaytestSubredditCreatePlaytestSubredditJSON(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        const body = JSON.parse(data.toString() || '{}');
        request = CreatePlaytestSubredditRequest.fromJSON(body);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the json request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.CreatePlaytestSubreddit(ctx, inputReq);
        });
    }
    else {
        response = await service.CreatePlaytestSubreddit(ctx, request);
    }
    return JSON.stringify(CreatePlaytestSubredditResponse.toJSON(response));
}
async function handlePlaytestSubredditCreatePlaytestSubredditProtobuf(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        request = CreatePlaytestSubredditRequest.decode(data);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the protobuf request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.CreatePlaytestSubreddit(ctx, inputReq);
        });
    }
    else {
        response = await service.CreatePlaytestSubreddit(ctx, request);
    }
    return Buffer.from(CreatePlaytestSubredditResponse.encode(response).finish());
}
