// @devvit/protos/json/devvit/dev_portal/app_version/info/app_version_info.js
/**
 * #app_version_info.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
export var VersionVisibility;
(function (VersionVisibility) {
    VersionVisibility[VersionVisibility["PUBLIC"] = 0] = "PUBLIC";
    VersionVisibility[VersionVisibility["PRIVATE"] = 1] = "PRIVATE";
    VersionVisibility[VersionVisibility["UNLISTED"] = 2] = "UNLISTED";
    VersionVisibility[VersionVisibility["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(VersionVisibility || (VersionVisibility = {}));
export var BuildStatus;
(function (BuildStatus) {
    BuildStatus[BuildStatus["BUILDING"] = 0] = "BUILDING";
    BuildStatus[BuildStatus["READY"] = 1] = "READY";
    BuildStatus[BuildStatus["FAILED"] = 2] = "FAILED";
    BuildStatus[BuildStatus["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BuildStatus || (BuildStatus = {}));
export var InstallationType;
(function (InstallationType) {
    InstallationType[InstallationType["USER"] = 0] = "USER";
    InstallationType[InstallationType["SUBREDDIT"] = 1] = "SUBREDDIT";
    InstallationType[InstallationType["SUBREDDIT_GLOBAL"] = 2] = "SUBREDDIT_GLOBAL";
    InstallationType[InstallationType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(InstallationType || (InstallationType = {}));
export var ComputePool;
(function (ComputePool) {
    ComputePool[ComputePool["LOW"] = 0] = "LOW";
    /** @deprecated */
    ComputePool[ComputePool["MEDIUM"] = 1] = "MEDIUM";
    ComputePool[ComputePool["HIGH"] = 2] = "HIGH";
    ComputePool[ComputePool["RESTRICTED"] = 3] = "RESTRICTED";
    ComputePool[ComputePool["INTERNAL"] = 4] = "INTERNAL";
    ComputePool[ComputePool["BETA"] = 5] = "BETA";
    ComputePool[ComputePool["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(ComputePool || (ComputePool = {}));
