// @devvit/protos/json/devvit/dev_portal/app/categories/categories.js
/**
 * #categories.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
/** Contains every valid category an App can have. */
export var Categories;
(function (Categories) {
    Categories[Categories["MODERATION"] = 0] = "MODERATION";
    Categories[Categories["CUSTOM_POSTS"] = 1] = "CUSTOM_POSTS";
    Categories[Categories["UTILITIES"] = 2] = "UTILITIES";
    Categories[Categories["OTHER_SITES"] = 3] = "OTHER_SITES";
    Categories[Categories["FUNNY"] = 4] = "FUNNY";
    Categories[Categories["WEIRD"] = 5] = "WEIRD";
    Categories[Categories["AWW"] = 6] = "AWW";
    Categories[Categories["LEARNING"] = 7] = "LEARNING";
    Categories[Categories["FINANCIAL"] = 8] = "FINANCIAL";
    Categories[Categories["SPORTS"] = 9] = "SPORTS";
    Categories[Categories["GAMES"] = 10] = "GAMES";
    Categories[Categories["MOVIES"] = 11] = "MOVIES";
    Categories[Categories["MUSIC"] = 12] = "MUSIC";
    Categories[Categories["TV"] = 13] = "TV";
    Categories[Categories["BOOKS"] = 14] = "BOOKS";
    Categories[Categories["ANIME"] = 15] = "ANIME";
    Categories[Categories["PROGRAMMING"] = 16] = "PROGRAMMING";
    Categories[Categories["COOKING"] = 17] = "COOKING";
    Categories[Categories["FITNESS"] = 18] = "FITNESS";
    Categories[Categories["TRAVEL"] = 19] = "TRAVEL";
    Categories[Categories["CONTEXT_MENU"] = 20] = "CONTEXT_MENU";
    Categories[Categories["MOD_CONTEXT"] = 21] = "MOD_CONTEXT";
    Categories[Categories["FILTER"] = 22] = "FILTER";
    Categories[Categories["POST_FILTER"] = 23] = "POST_FILTER";
    Categories[Categories["COMMENT_FILTER"] = 24] = "COMMENT_FILTER";
    Categories[Categories["ANTI_SPAM"] = 25] = "ANTI_SPAM";
    Categories[Categories["ANTI_HARASSMENT"] = 26] = "ANTI_HARASSMENT";
    Categories[Categories["NSFW_FILTER"] = 27] = "NSFW_FILTER";
    Categories[Categories["CITIZEN_FILTER"] = 28] = "CITIZEN_FILTER";
    Categories[Categories["POST_FLAIR"] = 29] = "POST_FLAIR";
    Categories[Categories["COMMENT_FLAIR"] = 30] = "COMMENT_FLAIR";
    Categories[Categories["USER_FLAIR"] = 31] = "USER_FLAIR";
    Categories[Categories["MESSAGING"] = 32] = "MESSAGING";
    Categories[Categories["MISCHIEF"] = 420] = "MISCHIEF";
    Categories[Categories["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(Categories || (Categories = {}));
