// @devvit/protos/json/devvit/plugin/logger/logger.js
/**
 * #logger.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
export var Severity;
(function (Severity) {
    /**
     * DEBUG - Debug logs should never be committed.
     *
     * These logs are always enabled in development builds but disabled in
     * production.
     */
    Severity[Severity["DEBUG"] = 0] = "DEBUG";
    /**
     * INFO - Informational logs should be used to mark significant non-erroneous events.
     * Little information is significant.
     *
     * These logs are enabled by default in development builds but may be
     * disabled in production.
     */
    Severity[Severity["INFO"] = 1] = "INFO";
    /**
     * WARN - Warnings should be reported as long as they do not contain sensitive data.
     *
     * These logs are always enabled.
     */
    Severity[Severity["WARN"] = 2] = "WARN";
    /**
     * ERROR - Errors should always be reported as long as they do not contain sensitive
     * data.
     */
    Severity[Severity["ERROR"] = 3] = "ERROR";
    /**
     * VERBOSE - Verbose logs are useful for tracing but usually a hindrance for others not
     * working where the log is placed. Keep our logs clean and add with care.
     *
     * These logs are disabled by default in all environments but can be enabled.
     */
    Severity[Severity["VERBOSE"] = 4] = "VERBOSE";
    Severity[Severity["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(Severity || (Severity = {}));
