// @devvit/protos/types/devvit/gateway/v1alpha/triggers_v2.js
/**
 * #triggers_v2.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
import _m0 from "protobufjs/minimal.js";
import { Empty } from "../../../google/protobuf/empty.js";
import { InstallationInfo } from "../../dev_portal/installation/installation.js";
export const TriggersV2ServiceName = "devvit.gateway.v1alpha.TriggersV2";
export class TriggersV2ClientImpl {
    constructor(rpc, opts) {
        this.service = opts?.service || TriggersV2ServiceName;
        this.rpc = rpc;
        this.OnInstall = this.OnInstall.bind(this);
        this.OnInstallChanged = this.OnInstallChanged.bind(this);
        this.OnUninstall = this.OnUninstall.bind(this);
        this.OnInstallSettingsChanged = this.OnInstallSettingsChanged.bind(this);
    }
    OnInstall(request, metadata) {
        const data = InstallationInfo.encode(request).finish();
        const promise = this.rpc.request(this.service, "OnInstall", data, metadata);
        return promise.then((data) => Empty.decode(_m0.Reader.create(data)));
    }
    OnInstallChanged(request, metadata) {
        const data = InstallationInfo.encode(request).finish();
        const promise = this.rpc.request(this.service, "OnInstallChanged", data, metadata);
        return promise.then((data) => Empty.decode(_m0.Reader.create(data)));
    }
    OnUninstall(request, metadata) {
        const data = InstallationInfo.encode(request).finish();
        const promise = this.rpc.request(this.service, "OnUninstall", data, metadata);
        return promise.then((data) => Empty.decode(_m0.Reader.create(data)));
    }
    OnInstallSettingsChanged(request, metadata) {
        const data = InstallationInfo.encode(request).finish();
        const promise = this.rpc.request(this.service, "OnInstallSettingsChanged", data, metadata);
        return promise.then((data) => Empty.decode(_m0.Reader.create(data)));
    }
}
export const TriggersV2Definition = {
    name: "TriggersV2",
    fullName: "devvit.gateway.v1alpha.TriggersV2",
    methods: {
        onInstall: {
            name: "OnInstall",
            requestType: InstallationInfo,
            requestStream: false,
            responseType: Empty,
            responseStream: false,
            options: {},
        },
        onInstallChanged: {
            name: "OnInstallChanged",
            requestType: InstallationInfo,
            requestStream: false,
            responseType: Empty,
            responseStream: false,
            options: {},
        },
        onUninstall: {
            name: "OnUninstall",
            requestType: InstallationInfo,
            requestStream: false,
            responseType: Empty,
            responseStream: false,
            options: {},
        },
        onInstallSettingsChanged: {
            name: "OnInstallSettingsChanged",
            requestType: InstallationInfo,
            requestStream: false,
            responseType: Empty,
            responseStream: false,
            options: {},
        },
    },
};
