// @devvit/protos/types/devvit/gateway/v1alpha/triggers_v2.twirp-client.js
import { InstallationInfo } from '../../dev_portal/installation/installation.js';
import { Empty } from '../../../google/protobuf/empty.js';
export class TriggersV2ClientJSON {
    constructor(rpc) {
        this.rpc = rpc;
        this.OnInstall.bind(this);
        this.OnInstallChanged.bind(this);
        this.OnUninstall.bind(this);
        this.OnInstallSettingsChanged.bind(this);
    }
    OnInstall(request) {
        const data = InstallationInfo.toJSON(request);
        const promise = this.rpc.request('devvit.gateway.v1alpha.TriggersV2', 'OnInstall', 'application/json', data);
        return promise.then((data) => Empty.fromJSON(data));
    }
    OnInstallChanged(request) {
        const data = InstallationInfo.toJSON(request);
        const promise = this.rpc.request('devvit.gateway.v1alpha.TriggersV2', 'OnInstallChanged', 'application/json', data);
        return promise.then((data) => Empty.fromJSON(data));
    }
    OnUninstall(request) {
        const data = InstallationInfo.toJSON(request);
        const promise = this.rpc.request('devvit.gateway.v1alpha.TriggersV2', 'OnUninstall', 'application/json', data);
        return promise.then((data) => Empty.fromJSON(data));
    }
    OnInstallSettingsChanged(request) {
        const data = InstallationInfo.toJSON(request);
        const promise = this.rpc.request('devvit.gateway.v1alpha.TriggersV2', 'OnInstallSettingsChanged', 'application/json', data);
        return promise.then((data) => Empty.fromJSON(data));
    }
}
export class TriggersV2ClientProtobuf {
    constructor(rpc) {
        this.rpc = rpc;
        this.OnInstall.bind(this);
        this.OnInstallChanged.bind(this);
        this.OnUninstall.bind(this);
        this.OnInstallSettingsChanged.bind(this);
    }
    OnInstall(request) {
        const data = InstallationInfo.encode(request).finish();
        const promise = this.rpc.request('devvit.gateway.v1alpha.TriggersV2', 'OnInstall', 'application/protobuf', data);
        return promise.then((data) => Empty.decode(data));
    }
    OnInstallChanged(request) {
        const data = InstallationInfo.encode(request).finish();
        const promise = this.rpc.request('devvit.gateway.v1alpha.TriggersV2', 'OnInstallChanged', 'application/protobuf', data);
        return promise.then((data) => Empty.decode(data));
    }
    OnUninstall(request) {
        const data = InstallationInfo.encode(request).finish();
        const promise = this.rpc.request('devvit.gateway.v1alpha.TriggersV2', 'OnUninstall', 'application/protobuf', data);
        return promise.then((data) => Empty.decode(data));
    }
    OnInstallSettingsChanged(request) {
        const data = InstallationInfo.encode(request).finish();
        const promise = this.rpc.request('devvit.gateway.v1alpha.TriggersV2', 'OnInstallSettingsChanged', 'application/protobuf', data);
        return promise.then((data) => Empty.decode(data));
    }
}
