// @devvit/protos/types/devvit/gateway/v1alpha/triggers_v2.twirp.js
import { TwirpServer, TwirpError, TwirpErrorCode, TwirpContentType, chainInterceptors, } from 'twirp-ts';
import { InstallationInfo } from '../../dev_portal/installation/installation.js';
import { Empty } from '../../../google/protobuf/empty.js';
export var TriggersV2Method;
(function (TriggersV2Method) {
    TriggersV2Method["OnInstall"] = "OnInstall";
    TriggersV2Method["OnInstallChanged"] = "OnInstallChanged";
    TriggersV2Method["OnUninstall"] = "OnUninstall";
    TriggersV2Method["OnInstallSettingsChanged"] = "OnInstallSettingsChanged";
})(TriggersV2Method || (TriggersV2Method = {}));
export const TriggersV2MethodList = [
    TriggersV2Method.OnInstall,
    TriggersV2Method.OnInstallChanged,
    TriggersV2Method.OnUninstall,
    TriggersV2Method.OnInstallSettingsChanged,
];
export function createTriggersV2Server(service) {
    return new TwirpServer({
        service,
        packageName: 'devvit.gateway.v1alpha',
        serviceName: 'TriggersV2',
        methodList: TriggersV2MethodList,
        matchRoute: matchTriggersV2Route,
    });
}
function matchTriggersV2Route(method, events) {
    switch (method) {
        case 'OnInstall':
            return async (ctx, service, data, interceptors) => {
                ctx = { ...ctx, methodName: 'OnInstall' };
                await events.onMatch(ctx);
                return handleTriggersV2OnInstallRequest(ctx, service, data, interceptors);
            };
        case 'OnInstallChanged':
            return async (ctx, service, data, interceptors) => {
                ctx = { ...ctx, methodName: 'OnInstallChanged' };
                await events.onMatch(ctx);
                return handleTriggersV2OnInstallChangedRequest(ctx, service, data, interceptors);
            };
        case 'OnUninstall':
            return async (ctx, service, data, interceptors) => {
                ctx = { ...ctx, methodName: 'OnUninstall' };
                await events.onMatch(ctx);
                return handleTriggersV2OnUninstallRequest(ctx, service, data, interceptors);
            };
        case 'OnInstallSettingsChanged':
            return async (ctx, service, data, interceptors) => {
                ctx = { ...ctx, methodName: 'OnInstallSettingsChanged' };
                await events.onMatch(ctx);
                return handleTriggersV2OnInstallSettingsChangedRequest(ctx, service, data, interceptors);
            };
        default:
            events.onNotFound();
            const msg = `no handler found`;
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
function handleTriggersV2OnInstallRequest(ctx, service, data, interceptors) {
    switch (ctx.contentType) {
        case TwirpContentType.JSON:
            return handleTriggersV2OnInstallJSON(ctx, service, data, interceptors);
        case TwirpContentType.Protobuf:
            return handleTriggersV2OnInstallProtobuf(ctx, service, data, interceptors);
        default:
            const msg = 'unexpected Content-Type';
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
function handleTriggersV2OnInstallChangedRequest(ctx, service, data, interceptors) {
    switch (ctx.contentType) {
        case TwirpContentType.JSON:
            return handleTriggersV2OnInstallChangedJSON(ctx, service, data, interceptors);
        case TwirpContentType.Protobuf:
            return handleTriggersV2OnInstallChangedProtobuf(ctx, service, data, interceptors);
        default:
            const msg = 'unexpected Content-Type';
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
function handleTriggersV2OnUninstallRequest(ctx, service, data, interceptors) {
    switch (ctx.contentType) {
        case TwirpContentType.JSON:
            return handleTriggersV2OnUninstallJSON(ctx, service, data, interceptors);
        case TwirpContentType.Protobuf:
            return handleTriggersV2OnUninstallProtobuf(ctx, service, data, interceptors);
        default:
            const msg = 'unexpected Content-Type';
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
function handleTriggersV2OnInstallSettingsChangedRequest(ctx, service, data, interceptors) {
    switch (ctx.contentType) {
        case TwirpContentType.JSON:
            return handleTriggersV2OnInstallSettingsChangedJSON(ctx, service, data, interceptors);
        case TwirpContentType.Protobuf:
            return handleTriggersV2OnInstallSettingsChangedProtobuf(ctx, service, data, interceptors);
        default:
            const msg = 'unexpected Content-Type';
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
async function handleTriggersV2OnInstallJSON(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        const body = JSON.parse(data.toString() || '{}');
        request = InstallationInfo.fromJSON(body);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the json request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.OnInstall(ctx, inputReq);
        });
    }
    else {
        response = await service.OnInstall(ctx, request);
    }
    return JSON.stringify(Empty.toJSON(response));
}
async function handleTriggersV2OnInstallChangedJSON(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        const body = JSON.parse(data.toString() || '{}');
        request = InstallationInfo.fromJSON(body);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the json request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.OnInstallChanged(ctx, inputReq);
        });
    }
    else {
        response = await service.OnInstallChanged(ctx, request);
    }
    return JSON.stringify(Empty.toJSON(response));
}
async function handleTriggersV2OnUninstallJSON(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        const body = JSON.parse(data.toString() || '{}');
        request = InstallationInfo.fromJSON(body);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the json request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.OnUninstall(ctx, inputReq);
        });
    }
    else {
        response = await service.OnUninstall(ctx, request);
    }
    return JSON.stringify(Empty.toJSON(response));
}
async function handleTriggersV2OnInstallSettingsChangedJSON(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        const body = JSON.parse(data.toString() || '{}');
        request = InstallationInfo.fromJSON(body);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the json request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.OnInstallSettingsChanged(ctx, inputReq);
        });
    }
    else {
        response = await service.OnInstallSettingsChanged(ctx, request);
    }
    return JSON.stringify(Empty.toJSON(response));
}
async function handleTriggersV2OnInstallProtobuf(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        request = InstallationInfo.decode(data);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the protobuf request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.OnInstall(ctx, inputReq);
        });
    }
    else {
        response = await service.OnInstall(ctx, request);
    }
    return Buffer.from(Empty.encode(response).finish());
}
async function handleTriggersV2OnInstallChangedProtobuf(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        request = InstallationInfo.decode(data);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the protobuf request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.OnInstallChanged(ctx, inputReq);
        });
    }
    else {
        response = await service.OnInstallChanged(ctx, request);
    }
    return Buffer.from(Empty.encode(response).finish());
}
async function handleTriggersV2OnUninstallProtobuf(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        request = InstallationInfo.decode(data);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the protobuf request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.OnUninstall(ctx, inputReq);
        });
    }
    else {
        response = await service.OnUninstall(ctx, request);
    }
    return Buffer.from(Empty.encode(response).finish());
}
async function handleTriggersV2OnInstallSettingsChangedProtobuf(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        request = InstallationInfo.decode(data);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the protobuf request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.OnInstallSettingsChanged(ctx, inputReq);
        });
    }
    else {
        response = await service.OnInstallSettingsChanged(ctx, request);
    }
    return Buffer.from(Empty.encode(response).finish());
}
