// @devvit/protos/json/devvit/runtime/bundle.js
/**
 * #bundle.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
/** See also LinkedBundle. */
export var LinkedBundleTargetPlatform;
(function (LinkedBundleTargetPlatform) {
    /** BROWSER - A LinkedBundle for browser runtimes. */
    LinkedBundleTargetPlatform[LinkedBundleTargetPlatform["BROWSER"] = 0] = "BROWSER";
    /** V8 - A LinkedBundle for V8 runtimes. */
    LinkedBundleTargetPlatform[LinkedBundleTargetPlatform["V8"] = 1] = "V8";
    /**
     * SKINNY - An API definition that cannot be executed. This is a LinkedBundle with no
     * code and no dependencies.
     */
    LinkedBundleTargetPlatform[LinkedBundleTargetPlatform["SKINNY"] = 2] = "SKINNY";
    LinkedBundleTargetPlatform[LinkedBundleTargetPlatform["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(LinkedBundleTargetPlatform || (LinkedBundleTargetPlatform = {}));
export var TargetRuntime;
(function (TargetRuntime) {
    /** UNIVERSAL - Server-only as of 2025-06-04 or so. */
    TargetRuntime[TargetRuntime["UNIVERSAL"] = 0] = "UNIVERSAL";
    TargetRuntime[TargetRuntime["CLIENT"] = 1] = "CLIENT";
    TargetRuntime[TargetRuntime["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(TargetRuntime || (TargetRuntime = {}));
